--- buildrootref/package/Config.in	2006-11-16 21:41:24.000000000 +0100
+++ buildroot/package/Config.in	2006-11-16 21:40:49.000000000 +0100
@@ -120,7 +120,6 @@ source "package/pppd/Config.in"
 source "package/procps/Config.in"
 source "package/psmisc/Config.in"
 source "package/python/Config.in"
-source "package/qte/Config.in"
 source "package/raidtools/Config.in"
 source "package/readline/Config.in"
 source "package/rsync/Config.in"
