--- buildrootref/package/Config.in	2007-11-14 21:43:29.000000000 +0100
+++ buildroot/package/Config.in	2007-11-14 21:44:15.000000000 +0100
@@ -223,6 +223,7 @@
 source "package/dm/Config.in"
 source "package/dmraid/Config.in"
 source "package/e2fsprogs/Config.in"
+source "package/fpgaregs/Config.in"
 source "package/gadgetfs-test/Config.in"
 source "package/hal/Config.in"
 if !BR2_PACKAGE_BUSYBOX_HIDE_OTHERS
