--- buildrootref/package/Config.in	2007-10-06 09:15:53.000000000 +0200
+++ buildroot/package/Config.in	2007-10-13 18:08:37.000000000 +0200
@@ -48,6 +48,7 @@
 source "package/dmalloc/Config.in"
 source "package/expat/Config.in"
 source "package/fakeroot/Config.in"
+source "package/fpgaregs/Config.in"
 source "package/gettext/Config.in"
 source "package/gmp/Config.in"
 source "package/mpfr/Config.in"
