--- buildrootref/package/Config.in	2007-01-21 23:30:06.000000000 +0100
+++ buildroot/package/Config.in	2007-02-12 16:32:53.000000000 +0100
@@ -126,6 +126,7 @@ source "package/readline/Config.in"
 source "package/rsync/Config.in"
 source "package/ruby/Config.in"
 source "package/rxvt/Config.in"
+source "package/samba/Config.in"
 source "package/sdl/Config.in"
 source "package/sfdisk/Config.in"
 source "package/slang/Config.in"
