--- buildroot.ori/package/sdl/Config.in	2007-03-03 21:28:52.000000000 +0100
+++ buildroot/package/sdl/Config.in	2007-03-02 21:22:13.000000000 +0100
@@ -16,7 +16,6 @@
 	  and supports the following formats: BMP, GIF, JPEG, LBM, PCX, PNG, PNM, TGA, TIFF, XCF, XPM, XV.
 
 	  http://www.libsdl.org/projects/SDL_image/
-	  
 
 config BR2_PACKAGE_SDL_TTF
 	bool "SDL_ttf"
@@ -27,4 +26,23 @@
 	  This is a sample library which allows you to use TrueType fonts in your SDL applications.
 
 	  http://www.libsdl.org/projects/SDL_ttf/
-	  
+
+config BR2_PACKAGE_SDL_MIXER
+	bool "SDL_Mixer"
+	depends on BR2_PACKAGE_SDL
+	default n
+	help
+
+	  SDL_mixer is a sample multi-channel audio mixer library. 
+
+	  http://www.libsdl.org/projects/SDL_mixer/
+
+config BR2_PACKAGE_SDL_NET
+	bool "SDL_Net"
+	depends on BR2_PACKAGE_SDL
+	default n
+	help
+
+	  This is a small sample cross-platform networking library, with a sample chat client and server application.
+
+	  http://www.libsdl.org/projects/SDL_net/
