--- buildroot.ori/package/Config.in	2007-03-03 21:28:52.000000000 +0100
+++ buildroot/package/Config.in	2007-03-02 21:57:27.000000000 +0100
@@ -118,6 +118,7 @@
 source "package/portage/Config.in"
 source "package/portmap/Config.in"
 source "package/pppd/Config.in"
+source "package/prboom/Config.in"
 source "package/procps/Config.in"
 source "package/psmisc/Config.in"
 source "package/python/Config.in"
