--- buildroot.ori/package/Config.in	2007-03-23 22:30:10.000000000 +0100
+++ buildroot/package/Config.in	2007-03-23 22:30:59.000000000 +0100
@@ -80,6 +80,7 @@
 source "package/libusb/Config.in"
 source "package/lighttpd/Config.in"
 source "package/links/Config.in"
+source "package/lmbench/Config.in"
 source "package/lrzsz/Config.in"
 source "package/lsof/Config.in"
 source "package/ltp-testsuite/Config.in"
