--- buildrootref/package/Config.in	2007-11-21 22:53:36.000000000 +0100
+++ buildroot/package/Config.in	2007-11-21 22:54:09.000000000 +0100
@@ -80,6 +80,7 @@
 source "package/liblockfile/Config.in"
 source "package/libsysfs/Config.in"
 source "package/libxml2/Config.in"
+source "package/lmbench/Config.in"
 source "package/lockfile-progs/Config.in"
 source "package/lsof/Config.in"
 source "package/ltp-testsuite/Config.in"
