--- buildroot/package/Config.in.ref     2008-11-03 16:03:10.000000000 +0100
+++ buildroot/package/Config.in 2008-11-03 16:07:18.000000000 +0100
@@ -88,6 +88,7 @@ source "package/libgpg-error/Config.in"
 source "package/libiconv/Config.in"
 source "package/liblockfile/Config.in"
 source "package/libsysfs/Config.in"
+source "package/lmbench/Config.in"
 source "package/lockfile-progs/Config.in"
 source "package/logrotate/Config.in"
 source "package/lsof/Config.in"
