--- buildrootref/package/Config.in	2007-10-13 18:34:18.000000000 +0200
+++ buildroot/package/Config.in	2007-10-13 18:36:53.000000000 +0200
@@ -85,6 +85,7 @@
 source "package/ltp-testsuite/Config.in"
 source "package/ltrace/Config.in"
 source "package/ltt/Config.in"
+source "package/lua/Config.in"
 if !BR2_PACKAGE_BUSYBOX_HIDE_OTHERS
 source "package/module-init-tools/Config.in"
 source "package/modutils/Config.in"
