--- buildroot/package/sdl/Config.in.old	2007-08-01 18:34:54.000000000 +0200
+++ buildroot/package/sdl/Config.in	2007-08-01 18:35:51.000000000 +0200
@@ -45,3 +45,13 @@
 	  This is a small sample cross-platform networking library, with a sample chat client and server application.
 
 	  http://www.libsdl.org/projects/SDL_net/
+
+config BR2_PACKAGE_SDL_WATER
+	bool "SDL_water"
+	depends on BR2_PACKAGE_SDL
+	default n
+	help
+
+	  Water effect demo coded with SDL
+
+	  http://www.libsdl.org/projects/SDL_water/
