diff -purN -x .svn -x '*~' buildrootref/target/device/armadeus/Config.in buildroot/target/device/armadeus/Config.in
--- buildrootref/target/device/armadeus/Config.in	2007-10-15 22:30:18.000000000 +0200
+++ buildroot/target/device/armadeus/Config.in	2007-10-14 23:35:28.000000000 +0200
@@ -68,7 +68,7 @@ config BR2_ARMADEUS_PACKAGE
 	source package/ch7024ctrl/Config.in
 	comment "---"    
     
-config BR2_PACKAGE_LINUX
+config BR2_PACKAGE_LINUX_ARM
 	bool "linux"
 	default n
 	depends BR2_TARGET_ARMADEUS
@@ -79,7 +79,7 @@ config BR2_PACKAGE_LINUX
 choice
 	prompt "Linux kernel version"
 	default BR2_PACKAGE_LINUX_VERSION_2_6_12
-	depends on BR2_PACKAGE_LINUX
+	depends on BR2_PACKAGE_LINUX_ARM
 	help
 	    Linux kernel version you want to use with your Armadeus board.
 	    If you plan to use MMC/SD cards then choose 2.6.18.1, otherwise 2.6.12
diff -purN -x .svn -x '*~' buildrootref/target/device/armadeus/linux/linux.mk buildroot/target/device/armadeus/linux/linux.mk
--- buildrootref/target/device/armadeus/linux/linux.mk	2007-10-15 22:30:17.000000000 +0200
+++ buildroot/target/device/armadeus/linux/linux.mk	2007-10-14 23:35:40.000000000 +0200
@@ -173,6 +173,6 @@ endif
 	-(cd $(BUILD_DIR); \
 	diff -purN -x '*~' ref/linux-$(LINUX_VERSION) linux-$(LINUX_VERSION) > $(BUILD_DIR)/../newlinux.diff)
 
-ifeq ($(strip $(BR2_PACKAGE_LINUX)),y)
+ifeq ($(strip $(BR2_PACKAGE_LINUX_ARM)),y)
 TARGETS+=linux
 endif
diff -purN -x .svn -x '*~' buildrootref/target/device/armadeus/Makefile.in buildroot/target/device/armadeus/Makefile.in
--- buildrootref/target/device/armadeus/Makefile.in	2007-10-15 22:30:18.000000000 +0200
+++ buildroot/target/device/armadeus/Makefile.in	2007-10-14 23:28:34.000000000 +0200
@@ -1,6 +1,7 @@
 # Override the default uClibc configuration
 ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
 
+
 ARMADEUS_PATH=target/device/armadeus
 ARMADEUS_TARGET_NAME=$(strip $(subst ",, $(BR2_TARGET_ARMADEUS_NAME)))
 ARMADEUS_CONFIG_PATH=$(ARMADEUS_PATH)/$(ARMADEUS_TARGET_NAME)
@@ -10,8 +11,11 @@ include target/device/armadeus/linux/lin
 include target/device/armadeus/rootfs/Makefile.in
 include target/device/armadeus/$(ARMADEUS_TARGET_NAME)/Makefile.in
 
+LINUX26_KCONFIG=$(ARMADEUS_CONFIG_PATH)/linux-$(BR2_PACKAGE_LINUX_VERSION).config
+LINUX26_VERSION=$(strip $(subst ",, $(BR2_PACKAGE_LINUX_VERSION)))
+DOWNLOAD_LINUX26_VERSION=$(strip $(subst ",, $(BR2_PACKAGE_LINUX_VERSION)))
 
-BR2_PACKAGE_BUSYBOX_CONFIG=$(ARMADEUS_PATH)/busybox.config
-UCLIBC_CONFIG_FILE=$(ARMADEUS_PATH)/uClibc.config.$(ARCH)
+#BR2_PACKAGE_BUSYBOX_CONFIG=$(ARMADEUS_PATH)/busybox.config
+#UCLIBC_CONFIG_FILE=$(ARMADEUS_PATH)/uClibc.config.$(ARCH)
 endif
 
diff -purN -x .svn -x '*~' buildrootref/target/device/armadeus/rootfs/Makefile.in buildroot/target/device/armadeus/rootfs/Makefile.in
--- buildrootref/target/device/armadeus/rootfs/Makefile.in	2007-10-15 22:30:18.000000000 +0200
+++ buildroot/target/device/armadeus/rootfs/Makefile.in	2007-10-15 21:20:14.000000000 +0200
@@ -1,5 +1,5 @@
 # Armadeus target skeleton stuff, may be overridden
-TARGET_SKELETON=$(ARMADEUS_PATH)/rootfs/skel.tar.gz
-TARGET_SKEL_DIR=$(ARMADEUS_PATH)/rootfs/target_skeleton
-TARGET_DEVICE_TABLE=$(ARMADEUS_PATH)/rootfs/device_table.txt
+TARGET_SKELETON_LINKS:=$(ARMADEUS_PATH)/rootfs/skel.tar.gz
+TARGET_SKELETON:=$(ARMADEUS_PATH)/rootfs/target_skeleton
+TARGET_DEVICE_TABLE:=$(ARMADEUS_PATH)/rootfs/device_table.txt
 
