--- buildrootref/package/qtopia4/qtopia4.mk	2007-11-20 22:43:16.000000000 +0100
+++ buildroot/package/qtopia4/qtopia4.mk	2007-11-20 22:39:53.000000000 +0100
@@ -211,7 +211,6 @@
 		-examplesdir /usr/share/qt4/examples \
 		-demosdir /usr/share/qt4/demos \
 		-fast \
-		-nomake examples \
 		-no-rpath \
 		$(QTOPIA4_QT3SUPPORT) \
 		$(QTOPIA4_TSLIB) \
@@ -223,6 +222,8 @@
 
 $(QTOPIA4_TARGET_DIR)/lib/libQtCore.so.$(QTOPIA4_VERSION): $(QTOPIA4_TARGET_DIR)/.configured
 	$(TARGET_CONFIGURE_OPTS) $(MAKE) -C $(QTOPIA4_TARGET_DIR) sub-src
+	$(TARGET_CONFIGURE_OPTS) $(MAKE) -C $(QTOPIA4_TARGET_DIR) sub-demos
+	$(TARGET_CONFIGURE_OPTS) $(MAKE) -C $(QTOPIA4_TARGET_DIR) sub-examples
 
 $(TARGET_DIR)/usr/lib/libQtCore.so.$(QTOPIA4_VERSION): $(QTOPIA4_TARGET_DIR)/lib/libQtCore.so.$(QTOPIA4_VERSION)
 	mkdir -p $(TARGET_DIR)/usr/lib/fonts
