diff -Naur buildroot.ori/package/ltp-testsuite/ltp-testsuite.mk buildroot/package/ltp-testsuite/ltp-testsuite.mk
--- buildroot.ori/package/ltp-testsuite/ltp-testsuite.mk	2007-10-27 09:15:33.000000000 +0200
+++ buildroot/package/ltp-testsuite/ltp-testsuite.mk	2007-11-25 20:22:27.000000000 +0100
@@ -15,7 +15,8 @@
 #
 LTP_PATCHES:=ltp-testsuite-generate-needs-bash.patch \
 	     ltp-testsuite-sh-is-not-C-code.patch \
-	     ltp-testsuite.patch
+	     ltp-testsuite.patch \
+	     ltp-testsuite-missing-open-max.patch
 
 ifeq ($(BR2_PTHREADS_NATIVE),y)
 LTP_PATCHES+=ltp-testsuite-enable-openposix-for-nptl.patch
