diff -Naur buildroot.ori/package/fakeroot/fakeroot.mk buildroot/package/fakeroot/fakeroot.mk
--- buildroot.ori/package/fakeroot/fakeroot.mk	2007-12-14 22:13:00.000000000 +0100
+++ buildroot/package/fakeroot/fakeroot.mk	2007-12-14 22:24:02.000000000 +0100
@@ -5,7 +5,7 @@
 #############################################################
 FAKEROOT_VERSION:=1.8.2
 FAKEROOT_SOURCE:=fakeroot_$(FAKEROOT_VERSION).tar.gz
-FAKEROOT_SITE:=http://ftp.debian.org/debian/pool/main/f/fakeroot
+FAKEROOT_SITE:=http://downloads.sourceforge.net/armadeus
 FAKEROOT_CAT:=$(ZCAT)
 FAKEROOT_SOURCE_DIR:=$(BUILD_DIR)/fakeroot-$(FAKEROOT_VERSION)
 FAKEROOT_DIR1:=$(TOOL_BUILD_DIR)/fakeroot-$(FAKEROOT_VERSION)-host
