Index: package/setserial/setserial.mk
===================================================================
--- buildroot.org/package/setserial/setserial.mk	(révision 22865)
+++ buildroot/package/setserial/setserial.mk	(copie de travail)
@@ -7,7 +7,7 @@
 SETSERIAL_PATCH_VERSION:=-44
 #SETSERIAL_PATCH_FILE:=
 SETSERIAL_SOURCE:=setserial_$(SETSERIAL_VERSION)$(SETSERIAL_PATCH_VERSION).tar.gz
-SETSERIAL_SITE:=http://ftp.debian.org/debian/pool/main/s/setserial/
+SETSERIAL_SITE:=http://security.ubuntu.com/ubuntu/pool/main/s/setserial/
 SETSERIAL_DIR:=$(BUILD_DIR)/setserial-$(SETSERIAL_VERSION)
 SETSERIAL_BINARY:=setserial
 SETSERIAL_TARGET_BINARY:=usr/bin/setserial
