--- buildroot.work/target/linux/Makefile.in.advanced	2008-11-14 10:33:55.000000000 +0100
+++ buildroot/target/linux/Makefile.in.advanced	2008-11-13 17:49:25.000000000 +0100
@@ -460,7 +460,9 @@
 
 linux26-source: $(DL_DIR)/$(LINUX26_SOURCE) $(LINUX26_PATCH_SOURCE_FILE)
 
-linux26-unpacked: $(LINUX26_DIR)/.patched
+linux26-unpacked: $(LINUX26_DIR)/.unpacked
+
+linux26-patched: $(LINUX26_DIR)/.patched
 
 linux26-config: host-sed $(LINUX26_DIR)/.configured
 
