--- buildroot/project/project.mk.ref	2008-11-03 09:15:03.000000000 +0100
+++ buildroot/project/project.mk	2009-01-01 16:54:02.000000000 +0100
@@ -40,10 +40,12 @@
 		$(SED) '/BR2_UCLIBC_CONFIG/d' $(PROJECT_FILE); \
 		echo "BR2_UCLIBC_CONFIG=\"$(LOCAL)/$(PROJECT)/uclibc-$(UCLIBC_VER).config\"" >> $(PROJECT_FILE); \
 	fi
-	if [ -f $(UBOOT_DIR)/include/configs/$(PROJECT).h ]; then \
+	if [ -f $(U_BOOT_DIR)/include/configs/$(BOARD_NAME).h ]; then \
 		mkdir -p $(LOCAL)/$(PROJECT)/u-boot; \
-		cp $(UBOOT_DIR)/include/configs/$(PROJECT).h \
-			$(LOCAL)/$(PROJECT)/u-boot/$(PROJECT).h; \
+		cp $(U_BOOT_DIR)/include/configs/$(BOARD_NAME).h \
+			$(LOCAL)/$(PROJECT)/u-boot/$(BOARD_NAME).h; \
+		$(SED) '/BR2_TARGET_U_BOOT_CONFIG_HEADER_FILE/d' $(PROJECT_FILE); \
+		echo "BR2_TARGET_U_BOOT_CONFIG_HEADER_FILE=\"$(LOCAL)/$(PROJECT)/u-boot/$(BOARD_NAME).h\"" >> $(PROJECT_FILE); \
 	fi
 
 getconfig: $(CONFIG)/conf
