diff -ruN buildroot/package/Config.in buildroot.new/package/Config.in
--- buildroot/package/Config.in	2009-03-09 16:57:56.000000000 +0100
+++ buildroot.new/package/Config.in	2009-02-24 11:33:56.000000000 +0100
@@ -121,6 +121,8 @@
 source "package/which/Config.in"
 endif
 
+source "package/xenomai/Config.in"
+
 source "package/database/Config.in"
 
 if !BR2_PACKAGE_BUSYBOX_HIDE_OTHERS
diff -urN buildroot/target/linux/Makefile.in.advanced buildroot.new/target/linux/Makefile.in.advanced
--- buildroot/target/linux/Makefile.in.advanced	2009-02-24 11:42:16.000000000 +0100
+++ buildroot.new/target/linux/Makefile.in.advanced	2009-02-24 11:40:07.000000000 +0100
@@ -278,6 +278,9 @@
 	toolchain/patch-kernel.sh $(LINUX26_DIR) package/openswan \
 		linux-$(LINUX26_VERSION)\*.patch{,.gz,.bz2}
 endif
+ifeq ($(BR2_PACKAGE_XENOMAI),y)
+	make -C ../buildroot xenomai-patch-kernel
+endif
 	touch $@
 
 $(LINUX26_DIR)/.patched.arch: $(LINUX26_DIR)/.patched
