--- buildrootref/target/ubifs/ubifsroot.mk	2008-11-03 00:15:09.000000000 -0800
+++ buildroot/target/ubifs/ubifsroot.mk	2009-06-02 14:39:18.000000000 -0700
@@ -5,8 +5,8 @@
 #############################################################
 #MKFS_UBIFS_VERSION=2582f128dad78591bc3adcc87c343c690bb82e61
 #MKFS_UBIFS_URL=http://git.infradead.org/users/dedekind/mkfs.ubifs.git?a=snapshot;h=$(MKFS_UBIFS_VERSION);sf=tgz
-MKFS_UBIFS_VERSION=v0.4
-MKFS_UBIFS_URL=http://git.infradead.org/users/dedekind/mkfs.ubifs.git?a=snapshot;h=refs/tags/mkfs.ubifs-$(MKFS_UBIFS_VERSION);sf=tgz
+MKFS_UBIFS_VERSION=HEAD
+MKFS_UBIFS_URL=http://git.infradead.org/users/ahunter/mkfs.ubifs.git?a=snapshot;h=$(MKFS_UBIFS_VERSION);sf=tgz
 MKFS_UBIFS_SOURCE:=mkfs.ubifs-$(MKFS_UBIFS_VERSION).tar.gz
 MKFS_UBIFS_DIR:= $(BUILD_DIR)/mkfs-ubifs-$(MKFS_UBIFS_VERSION)
 MKFS_UBIFS_CAT:=$(ZCAT)
@@ -107,6 +107,13 @@ endif
 	chmod a+x $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	$(STAGING_DIR)/usr/bin/fakeroot -- $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	-@rm -f $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
+	cp target/device/armadeus/$(BR2_BOARD_NAME)/ubinize.cfg .
+	cp $(BINARIES_DIR)/$(BR2_BOARD_NAME)-rootfs.arm.ubifs .
+	ubinize -o system_ubi.img -m $(BR2_TARGET_ROOTFS_UBIFS_MINIOSIZE) -p 128KiB -s 512 ubinize.cfg
+	cp -f system_ubi.img $(BINARIES_DIR)/$(BR2_BOARD_NAME)-rootfs.arm.ubifs
+	rm system_ubi.img
+	rm $(BR2_BOARD_NAME)-rootfs.arm.ubifs
+	rm ubinize.cfg
 
 ifneq ($(UBIFS_ROOTFS_COMPRESSOR),)
 $(UBIFS_BASE).$(UBIFS_ROOTFS_COMPRESSOR_EXT): $(UBIFS_ROOTFS_COMPRESSOR_PREREQ) $(UBIFS_BASE)
--- buildrootref/target/ubifs/mkfs-ubifs-head.patch	1970-01-01 01:00:00.000000000 +0100
+++ buildroot/target/ubifs/mkfs-ubifs-head.patch	2009-06-07 11:57:11.000000000 +0200
@@ -0,0 +1,43 @@
+diff -purN mkfs-ubifs-HEAD/mkfs.ubifs.h mkfs-ubifs-HEAD.new/mkfs.ubifs.h
+--- mkfs-ubifs-HEAD/mkfs.ubifs.h	2008-08-12 10:34:49.000000000 +0200
++++ mkfs-ubifs-HEAD.new/mkfs.ubifs.h	2009-06-07 11:53:18.000000000 +0200
+@@ -35,6 +35,7 @@
+ #include <byteswap.h>
+ #include <linux/types.h>
+ #include <linux/fs.h>
++#include <linux/ext2_fs.h>
+ 
+ #include <getopt.h>
+ #include <sys/types.h>
+@@ -70,6 +71,31 @@
+ #error MKFS_UBIFS_COMPR_ZLIB != UBIFS_COMPR_ZLIB
+ #endif
+ 
++#ifndef FS_COMPR_FL
++#define FS_COMPR_FL EXT2_COMPR_FL
++#endif
++
++#ifndef FS_SYNC_FL
++#define FS_SYNC_FL EXT2_SYNC_FL
++#endif
++
++#ifndef FS_IMMUTABLE_FL
++#define FS_IMMUTABLE_FL EXT2_IMMUTABLE_FL
++#endif
++
++#ifndef FS_APPEND_FL
++#define FS_APPEND_FL EXT2_APPEND_FL
++#endif
++
++#ifndef FS_DIRSYNC_FL
++#define FS_DIRSYNC_FL EXT2_DIRSYNC_FL
++#endif
++
++#ifndef FS_IOC_GETFLAGS
++#define FS_IOC_GETFLAGS EXT2_IOC_GETFLAGS
++#endif
++
++
+ extern int verbose;
+ extern int debug_level;
+ 
