diff -u a/Makefile.in b/Makefile.in
--- a/Makefile.in	2009-03-12 15:16:41.000000000 +0100
+++ b/Makefile.in	2009-03-12 15:16:54.000000000 +0100
@@ -228,7 +228,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I config
-SUDO = sudo
+SUDO = fakeroot
 SUBDIRS = src include config @XENO_MAYBE_DOCDIR@ scripts
 EXTRA_DIST = CREDITS README.INSTALL TROUBLESHOOTING ksrc examples debian
 all: all-recursive
