--- a/arch/arm/mach-imx/include/mach/gpio.h	2009-04-03 15:28:59.000000000 +0200
+++ b/arch/arm/mach-imx/include/mach/gpio.h	2009-04-03 15:29:11.000000000 +0200
@@ -28,14 +28,10 @@
 
 static inline void imx_gpio_set_value_inline(unsigned gpio, int value)
 {
-	unsigned long flags;
-
-	raw_local_irq_save(flags);
 	if(value)
 		DR(gpio >> GPIO_PORT_SHIFT) |= (1 << (gpio & GPIO_PIN_MASK));
 	else
 		DR(gpio >> GPIO_PORT_SHIFT) &= ~(1 << (gpio & GPIO_PIN_MASK));
-	raw_local_irq_restore(flags);
 }
 
 static inline void imx_gpio_set_value(unsigned gpio, int value)
