--- linux-2.6.18/scripts/mod/sumversion.c	2006-09-20 05:42:06.000000000 +0200
+++ sumversion.c	2006-10-07 16:42:31.609375000 +0200
@@ -7,6 +7,7 @@
 #include <ctype.h>
 #include <errno.h>
 #include <string.h>
+#include <limits.h>
 #include "modpost.h"
 
 /*
--- linux-2.6.18/scripts/mod/file2alias.c	2006-09-20 05:42:06.000000000 +0200
+++ file2alias.c	2006-10-07 15:43:32.890625000 +0200
@@ -11,7 +11,6 @@
  */
 
 #include "modpost.h"
-
 /* We use the ELF typedefs for kernel_ulong_t but bite the bullet and
  * use either stdint.h or inttypes.h for the rest. */
 #if KERNEL_ELFCLASS == ELFCLASS32
@@ -26,18 +25,19 @@ typedef Elf64_Addr	kernel_ulong_t;
 #else
 #include <stdint.h>
 #endif
-
+#include "../../include/linux/input.h"
 #include <ctype.h>
 
+/*
 typedef uint32_t	__u32;
 typedef uint16_t	__u16;
 typedef unsigned char	__u8;
-
+*/
 /* Big exception to the "don't include kernel headers into userspace, which
  * even potentially has different endianness and word sizes, since
  * we handle those differences explicitly below */
 #include "../../include/linux/mod_devicetable.h"
-#include "../../include/linux/input.h"
+
 
 #define ADD(str, sep, cond, field)                              \
 do {                                                            \
