diff -urN gpm-1.20.0/src/gpm-root.y gpm-1.20.0.mod/src/gpm-root.y
--- gpm-1.20.0/src/gpm-root.y   2002-02-23 16:42:23.000000000 +0100
+++ gpm-1.20.0.mod/src/gpm-root.y       2007-06-29 16:17:59.000000000 +0200
@@ -332,6 +332,7 @@
          case ' ': /* fall through */
          case '\t': continue;
          default: if (!isalpha(c)) return(c);
+         break;
       }
       /* get a single word and convert it */
       do {
@@ -498,6 +499,7 @@
       case F_CREATE:
       case F_POST: break;
       case F_INVOKE: /* MISS */
+      break;
    }
    return 0;
 }
diff -urN gpm-1.20.0/src/synaptics.c gpm-1.20.0.mod/src/synaptics.c
--- gpm-1.20.0/src/synaptics.c  2002-02-23 16:42:23.000000000 +0100
+++ gpm-1.20.0.mod/src/synaptics.c      2007-06-29 16:16:13.000000000 +0200
@@ -816,6 +816,7 @@
 	      break;
 
 	    default:
+	      break;
 	    }
 	  }
 	}
