diff -purN buildroot-ref/.defconfig buildroot/.defconfig
--- buildroot-ref/.defconfig	2005-11-29 09:15:40.000000000 +0100
+++ buildroot/.defconfig	2005-12-24 12:45:44.000000000 +0100
@@ -2,55 +2,328 @@
 # Automatically generated make config: don't edit
 #
 BR2_HAVE_DOT_CONFIG=y
-# BR2_arm is not set
+# BR2_alpha is not set
+BR2_arm=y
 # BR2_armeb is not set
 # BR2_cris is not set
-BR2_i386=y
+# BR2_i386 is not set
 # BR2_m68k is not set
 # BR2_mips is not set
 # BR2_mipsel is not set
 # BR2_powerpc is not set
 # BR2_sh is not set
 # BR2_sparc is not set
-BR2_ARCH="i386"
+# BR2_x86_64 is not set
+BR2_ARCH="arm"
+BR2_ENDIAN="LITTLE"
+
+#
+# Build options
+#
 BR2_WGET="wget --passive-ftp"
+BR2_SVN="svn co"
+# BR2_TAR_VERBOSITY is not set
+BR2_DL_DIR="$(BASE_DIR)/dl"
+BR2_SOURCEFORGE_MIRROR="easynews"
+BR2_STAGING_DIR="$(BUILD_DIR)/staging_dir"
+BR2_TOPDIR_PREFIX=""
+BR2_TOPDIR_SUFFIX=""
+BR2_JLEVEL=2
 
 #
 # Toolchain Options
 #
-BR2_KERNEL_HEADERS_2_4=y
-# BR2_KERNEL_HEADERS_2_6 is not set
-BR2_DEFAULT_KERNEL_HEADERS="2.4.27"
-BR2_UCLIBC_VERSION_SNAPSHOT=y
-BR2_USE_UCLIBC_SNAPSHOT="snapshot"
+
+#
+# Kernel Header Options
+#
+# BR2_KERNEL_HEADERS_2_4_25 is not set
+# BR2_KERNEL_HEADERS_2_4_27 is not set
+# BR2_KERNEL_HEADERS_2_4_29 is not set
+# BR2_KERNEL_HEADERS_2_4_31 is not set
+# BR2_KERNEL_HEADERS_2_6_11 is not set
+BR2_KERNEL_HEADERS_2_6_12=y
+BR2_DEFAULT_KERNEL_HEADERS="2.6.12"
+
+#
+# uClibc Options
+#
+# BR2_UCLIBC_VERSION_SNAPSHOT is not set
 # BR2_ENABLE_LOCALE is not set
-# BR2_BINUTILS_VERSION_2_14_90_0_6 is not set
-# BR2_BINUTILS_VERSION_2_14_90_0_7 is not set
+BR2_PTHREADS=y
+# BR2_PTHREADS_NATIVE is not set
+
+#
+# Binutils Options
+#
 # BR2_BINUTILS_VERSION_2_14_90_0_8 is not set
 # BR2_BINUTILS_VERSION_2_15 is not set
-# BR2_BINUTILS_VERSION_2_15_90_0_1 is not set
-# BR2_BINUTILS_VERSION_2_15_90_0_1_1 is not set
-# BR2_BINUTILS_VERSION_2_15_90_0_2 is not set
-# BR2_BINUTILS_VERSION_2_15_90_0_3 is not set
-# BR2_BINUTILS_VERSION_2_15_91_0_1 is not set
-BR2_BINUTILS_VERSION_2_15_91_0_2=y
-# BR2_BINUTILS_VERSION_2_15_92_0_2 is not set
-BR2_BINUTILS_VERSION="2.15.91.0.2"
-# BR2_GCC_VERSION_2_95 is not set
-# BR2_GCC_VERSION_3_3_3 is not set
-# BR2_GCC_VERSION_3_3_4 is not set
-# BR2_GCC_VERSION_3_4_0 is not set
-# BR2_GCC_VERSION_3_4_1 is not set
+# BR2_BINUTILS_VERSION_2_15_94_0_2_2 is not set
+# BR2_BINUTILS_VERSION_2_15_97 is not set
+# BR2_BINUTILS_VERSION_2_16 is not set
+BR2_BINUTILS_VERSION_2_16_1=y
+# BR2_BINUTILS_VERSION_2_16_90_0_3 is not set
+# BR2_BINUTILS_VERSION_2_16_91_0_2 is not set
+# BR2_BINUTILS_VERSION_2_16_91_0_3 is not set
+# BR2_BINUTILS_VERSION_2_16_91_0_4 is not set
+BR2_BINUTILS_VERSION="2.16.1"
+
+#
+# Gcc Options
+#
+# BR2_GCC_VERSION_3_3_5 is not set
+# BR2_GCC_VERSION_3_3_6 is not set
 BR2_GCC_VERSION_3_4_2=y
+# BR2_GCC_VERSION_3_4_3 is not set
+# BR2_GCC_VERSION_3_4_4 is not set
+# BR2_GCC_VERSION_4_0_0 is not set
+# BR2_GCC_VERSION_4_0_1 is not set
+# BR2_GCC_VERSION_4_0_2 is not set
+# BR2_GCC_VERSION_4_1_0 is not set
 BR2_GCC_VERSION="3.4.2"
-BR2_GCC_USE_SJLJ_EXCEPTIONS="--enable-sjlj-exceptions"
+# BR2_GCC_USE_SJLJ_EXCEPTIONS is not set
 BR2_EXTRA_GCC_CONFIG_OPTIONS=""
 # BR2_INSTALL_LIBSTDCPP is not set
-BR2_HOST_CCACHE=y
-BR2_PACKAGE_CCACHE=y
-# BR2_PACKAGE_GDB is not set
-# BR2_PACKAGE_GDB_SERVER is not set
-# BR2_HOST_GDB is not set
+# BR2_INSTALL_OBJC is not set
+
+#
+# Ccache Options
+#
+BR2_CCACHE=y
+
+#
+# Gdb Options
+#
+BR2_PACKAGE_GDB=y
+BR2_PACKAGE_GDB_SERVER=y
+# BR2_PACKAGE_GDB_CLIENT is not set
+# BR2_GDB_VERSION_6_2_1 is not set
+BR2_GDB_VERSION_6_3=y
+# BR2_GDB_VERSION_SNAPSHOT is not set
+BR2_GDB_VERSION="6.3"
+# BR2_PACKAGE_SSTRIP_TARGET is not set
+BR2_PACKAGE_SSTRIP_HOST=y
+
+#
+# Common Toolchain Options
+#
 BR2_ENABLE_MULTILIB=y
 BR2_LARGEFILE=y
+BR2_SOFT_FLOAT=y
 BR2_TARGET_OPTIMIZATION="-Os -pipe"
+BR2_CROSS_TOOLCHAIN_TARGET_UTILS=y
+
+#
+# Package Selection for the target
+#
+
+#
+# The default minimal system
+#
+BR2_PACKAGE_BUSYBOX=y
+# BR2_PACKAGE_BUSYBOX_SNAPSHOT is not set
+BR2_PACKAGE_BUSYBOX_INSTALL_SYMLINKS=y
+BR2_PACKAGE_BUSYBOX_CONFIG="package/busybox/busybox.config"
+
+#
+# The minimum needed to build a uClibc development system
+#
+# BR2_PACKAGE_BASH is not set
+# BR2_PACKAGE_BZIP2 is not set
+# BR2_PACKAGE_COREUTILS is not set
+# BR2_PACKAGE_DIFFUTILS is not set
+# BR2_PACKAGE_ED is not set
+# BR2_PACKAGE_FINDUTILS is not set
+# BR2_PACKAGE_FLEX is not set
+# BR2_PACKAGE_GAWK is not set
+# BR2_PACKAGE_GCC_TARGET is not set
+# BR2_PACKAGE_CCACHE_TARGET is not set
+# BR2_PACKAGE_GREP is not set
+# BR2_PACKAGE_MAKE is not set
+# BR2_PACKAGE_PATCH is not set
+# BR2_PACKAGE_SED is not set
+# BR2_PACKAGE_TAR is not set
+
+#
+# Other stuff
+#
+# BR2_PACKAGE_ACPID is not set
+# BR2_PACKAGE_AT is not set
+# BR2_PACKAGE_AUTOCONF is not set
+# BR2_PACKAGE_AUTOMAKE is not set
+# BR2_PACKAGE_BERKELEYDB is not set
+# BR2_PACKAGE_BISON is not set
+# BR2_PACKAGE_BOA is not set
+# BR2_PACKAGE_BRIDGE is not set
+# BR2_PACKAGE_CUSTOMIZE is not set
+# BR2_PACKAGE_ISC_DHCP is not set
+# BR2_PACKAGE_DIRECTFB is not set
+# BR2_PACKAGE_DISTCC is not set
+# BR2_PACKAGE_DM is not set
+# BR2_PACKAGE_DNSMASQ is not set
+# BR2_PACKAGE_DROPBEAR is not set
+# BR2_PACKAGE_E2FSPROGS is not set
+# BR2_PACKAGE_FAKEROOT is not set
+# BR2_PACKAGE_FILE is not set
+# BR2_PACKAGE_FREETYPE is not set
+# BR2_PACKAGE_GETTEXT is not set
+# BR2_PACKAGE_GZIP is not set
+# BR2_PACKAGE_HOSTAP is not set
+# BR2_PACKAGE_HOTPLUG is not set
+# BR2_PACKAGE_IOSTAT is not set
+# BR2_PACKAGE_IPROUTE2 is not set
+# BR2_PACKAGE_IPTABLES is not set
+# BR2_PACKAGE_JPEG is not set
+BR2_PACKAGE_LESS=y
+# BR2_PACKAGE_LIBELF is not set
+# BR2_PACKAGE_LIBFLOAT is not set
+# BR2_PACKAGE_LIBGLIB12 is not set
+# BR2_PACKAGE_LIBMAD is not set
+# BR2_PACKAGE_LIBPCAP is not set
+# BR2_PACKAGE_LIBPNG is not set
+# BR2_PACKAGE_LIBSYSFS is not set
+# BR2_PACKAGE_LIBTOOL is not set
+# BR2_PACKAGE_LINKS is not set
+# BR2_PACKAGE_LRZSZ is not set
+# BR2_PACKAGE_LTP-TESTSUITE is not set
+# BR2_PACKAGE_LTT is not set
+# BR2_PACKAGE_LVM2 is not set
+# BR2_PACKAGE_LZO is not set
+# BR2_PACKAGE_M4 is not set
+# BR2_PACKAGE_MDADM is not set
+# BR2_PACKAGE_MEMTESTER is not set
+# BR2_PACKAGE_MICROCOM is not set
+# BR2_PACKAGE_MICROPERL is not set
+# BR2_PACKAGE_MICROWIN is not set
+# BR2_PACKAGE_MKDOSFS is not set
+BR2_PACKAGE_MODULE_INIT_TOOLS=y
+# BR2_PACKAGE_MODUTILS is not set
+# BR2_PACKAGE_MPG123 is not set
+# BR2_PACKAGE_MROUTED is not set
+BR2_PACKAGE_MTD=y
+BR2_PACKAGE_MTD_FLASH_ERASE=y
+BR2_PACKAGE_MTD_FLASH_ERASEASLL=y
+BR2_PACKAGE_MTD_FLASH_INFO=y
+BR2_PACKAGE_MTD_FLASH_LOCK=y
+BR2_PACKAGE_MTD_FLASH_UNLOCK=y
+BR2_PACKAGE_MTD_FLASHCP=y
+BR2_PACKAGE_MTD_JFFS2DUMP=y
+BR2_PACKAGE_MTD_SUMTOOL=y
+BR2_PACKAGE_MTD_FTL_CHECK=y
+BR2_PACKAGE_MTD_FTL_FORMAT=y
+BR2_PACKAGE_MTD_NFTL_FORMAT=y
+BR2_PACKAGE_MTD_NFTLDUMP=y
+BR2_PACKAGE_MTD_MKFSJFFS2=y
+BR2_PACKAGE_MTD_MKFSJFFS=y
+BR2_PACKAGE_MTD_NANDDUMP=y
+BR2_PACKAGE_MTD_NANDWRITE=y
+BR2_PACKAGE_MTD_MTD_DEBUG=y
+BR2_PACKAGE_MTD_DOCFDISK=y
+BR2_PACKAGE_MTD_DOC_LOADBIOS=y
+BR2_PACKAGE_NANO=y
+BR2_PACKAGE_NCURSES=y
+# BR2_PACKAGE_NCURSES_TARGET_HEADERS is not set
+# BR2_PACKAGE_NETKITBASE is not set
+# BR2_PACKAGE_NETKITTELNET is not set
+# BR2_PACKAGE_NETSNMP is not set
+# BR2_PACKAGE_NEWT is not set
+# BR2_PACKAGE_NTP is not set
+# BR2_PACKAGE_OPENNTPD is not set
+# BR2_PACKAGE_OPENSSH is not set
+# BR2_PACKAGE_OPENSSL is not set
+# BR2_PACKAGE_OPENVPN is not set
+# BR2_PACKAGE_PCIUTILS is not set
+# BR2_PACKAGE_PCMCIA is not set
+# BR2_PACKAGE_PORTAGE is not set
+BR2_PACKAGE_PORTMAP=y
+# BR2_PACKAGE_PPPD is not set
+# BR2_PACKAGE_PROCPS is not set
+# BR2_PACKAGE_PSMISC is not set
+# BR2_PACKAGE_PYTHON is not set
+# BR2_PACKAGE_QTE is not set
+BR2_QTE_TMAKE_VERSION="1.13"
+# BR2_PACKAGE_RAIDTOOLS is not set
+# BR2_READLINE is not set
+# BR2_PACKAGE_RSYNC is not set
+# BR2_PACKAGE_RXVT is not set
+# BR2_PACKAGE_SDL is not set
+# BR2_PACKAGE_SFDISK is not set
+# BR2_PACKAGE_SLANG is not set
+# BR2_PACKAGE_SOCAT is not set
+# BR2_PACKAGE_STRACE is not set
+# BR2_PACKAGE_SYSKLOGD is not set
+# BR2_PACKAGE_TCL is not set
+# BR2_PACKAGE_TCPDUMP is not set
+# BR2_PACKAGE_TFTPD is not set
+# BR2_PACKAGE_TINYLOGIN is not set
+# BR2_PACKAGE_TINYX is not set
+# BR2_PACKAGE_TN5250 is not set
+# BR2_PACKAGE_TTCP is not set
+# BR2_PACKAGE_UDEV is not set
+# BR2_PACKAGE_UDHCP is not set
+BR2_PACKAGE_UTIL-LINUX=y
+# BR2_PACKAGE_VALGRIND is not set
+# BR2_PACKAGE_VTUN is not set
+# BR2_PACKAGE_WGET is not set
+# BR2_PACKAGE_WHICH is not set
+# BR2_PACKAGE_WIRELESS_TOOLS is not set
+# BR2_PACKAGE_XFSPROGS is not set
+# BR2_PACKAGE_XORG is not set
+BR2_PACKAGE_ZLIB=y
+# BR2_PACKAGE_ZLIB_TARGET_HEADERS is not set
+
+#
+# Target Options
+#
+
+#
+# filesystem for target device
+#
+# BR2_TARGET_ROOTFS_CRAMFS is not set
+# BR2_TARGET_ROOTFS_EXT2 is not set
+BR2_TARGET_ROOTFS_JFFS2=y
+BR2_TARGET_ROOTFS_JFFS2_EBSIZE=0x20000
+BR2_TARGET_ROOTFS_JFFS2_PAD=y
+BR2_TARGET_ROOTFS_JFFS2_PADSIZE=0x0
+BR2_TARGET_ROOTFS_JFFS2_LE=y
+# BR2_TARGET_ROOTFS_JFFS2_BE is not set
+# BR2_TARGET_ROOTFS_JFFS2_SQUASH is not set
+BR2_TARGET_ROOTFS_JFFS2_OUTPUT="$(IMAGE).jffs2"
+BR2_TARGET_ROOTFS_JFFS2_COPYTO=""
+# BR2_TARGET_ROOTFS_SQUASHFS is not set
+BR2_TARGET_ROOTFS_TAR=y
+BR2_TARGET_ROOTFS_TAR_OPTIONS=""
+
+#
+# bootloader for target device
+#
+
+#
+# Board Support Options
+#
+BR2_PACKAGE_LINUX=y
+
+#
+# Armadeus project Specific Device Support
+#
+BR2_TARGET_ARMADEUS=y
+# BR2_TARGET_ARMADEUS_APF9328 is not set
+BR2_TARGET_ARMADEUS_APM9328=y
+BR2_TARGET_ARMADEUS_NAME="apm9328"
+# BR2_TARGET_ARMADEUS_SDRAM_8 is not set
+# BR2_TARGET_ARMADEUS_SDRAM_16 is not set
+BR2_TARGET_ARMADEUS_SDRAM_32=y
+# BR2_TARGET_ARMADEUS_SDRAM_64 is not set
+BR2_TARGET_ARMADEUS_SDRAM_SIZE="32"
+
+#
+# Package support
+#
+BR2_TARGET_U-BOOT=y
+
+#
+# Generic System Support
+#
+# BR2_TARGET_GENERIC_ACCESS_POINT is not set
+# BR2_TARGET_GENERIC_FIREWALL is not set
+# BR2_TARGET_GENERIC_DEV_SYSTEM is not set
diff -purN buildroot-ref/package/module-init-tools/Config.in buildroot/package/module-init-tools/Config.in
--- buildroot-ref/package/module-init-tools/Config.in	2005-11-29 09:15:30.000000000 +0100
+++ buildroot/package/module-init-tools/Config.in	2005-12-21 16:39:18.000000000 +0100
@@ -7,4 +7,4 @@ config BR2_PACKAGE_MODULE_INIT_TOOLS
 	  (versions 2.5.48 and above). It serves the same function that 
 	  the "modutils" package serves for Linux 2.4.
 
-	  http://www.kernel.org/pub/linux/utils/kernel/module-init-tools/
+	  http://www.kernel.org/pub/linux/utils/kernel/module-init-tools/old/
diff -purN buildroot-ref/package/module-init-tools/module-init-tools.mk buildroot/package/module-init-tools/module-init-tools.mk
--- buildroot-ref/package/module-init-tools/module-init-tools.mk	2005-11-29 09:15:30.000000000 +0100
+++ buildroot/package/module-init-tools/module-init-tools.mk	2005-12-21 16:39:18.000000000 +0100
@@ -4,7 +4,7 @@
 #
 #############################################################
 MODULE_INIT_TOOLS_SOURCE=module-init-tools-3.1.tar.bz2
-MODULE_INIT_TOOLS_SITE=ftp://ftp.kernel.org/pub/linux/utils/kernel/module-init-tools/
+MODULE_INIT_TOOLS_SITE=ftp://ftp.kernel.org/pub/linux/utils/kernel/module-init-tools/old/
 MODULE_INIT_TOOLS_DIR=$(BUILD_DIR)/module-init-tools-3.1
 MODULE_INIT_TOOLS_BINARY=modprobe
 MODULE_INIT_TOOLS_TARGET_BINARY=$(TARGET_DIR)/sbin/$(MODULE_INIT_TOOLS_BINARY)
diff -purN buildroot-ref/target/device/Config.in buildroot/target/device/Config.in
--- buildroot-ref/target/device/Config.in	2005-11-29 09:15:23.000000000 +0100
+++ buildroot/target/device/Config.in	2005-12-24 12:30:36.000000000 +0100
@@ -4,6 +4,7 @@ source "target/device/Soekris/Config.in"
 source "target/device/Hitachi/Config.in"
 source "target/device/AMD/Config.in"
 source "target/device/Via/Config.in"
+source "target/device/armadeus/Config.in"
 
 # This must be last
 source "target/generic/Config.in"
diff -purN buildroot-ref/target/jffs2/jffs2root.mk buildroot/target/jffs2/jffs2root.mk
--- buildroot-ref/target/jffs2/jffs2root.mk	2005-11-29 09:15:23.000000000 +0100
+++ buildroot/target/jffs2/jffs2root.mk	2005-12-22 17:12:16.000000000 +0100
@@ -38,7 +38,7 @@ JFFS2_TARGET := $(subst ",,$(BR2_TARGET_
 # value of MKFS_JFFS2 to either the previously installed copy or the one
 # just built.
 #
-$(JFFS2_TARGET): host-fakeroot makedevs $(STAGING_DIR)/fakeroot.env mtd-host
+$(JFFS2_TARGET): host-fakeroot makedevs $(STAGING_DIR)/fakeroot.env mtd-host linux
 	-@find $(TARGET_DIR) -type f -perm +111 | xargs $(STRIP) 2>/dev/null || true;
 	@rm -rf $(TARGET_DIR)/usr/man
 	@rm -rf $(TARGET_DIR)/usr/share/man
