--- buildroot/package/sed/sed.mk	2005-11-29 09:15:28.000000000 +0100
+++ software/buildroot/package/sed/sed.mk	2006-01-27 00:44:03.000000000 +0100
@@ -16,7 +16,7 @@ SED_CPPFLAGS=-D_FILE_OFFSET_BITS=64
 endif
 #HOST_SED_DIR:=$(STAGING_DIR)
 HOST_SED_DIR:=$(TOOL_BUILD_DIR)
-SED:=$(HOST_SED_DIR)/bin/sed -i -e
+SED:=$(HOST_SED_DIR)/bin/sed -i "" -e
 HOST_SED_TARGET=$(shell package/sed/sedcheck.sh)
 
 $(DL_DIR)/$(SED_SOURCE):
Binary files buildroot/target/device/armadeus/u-boot/.DS_Store and software/buildroot/target/device/armadeus/u-boot/.DS_Store differ
diff -purN -x .svn -x '*~' buildroot/target/device/armadeus/u-boot/u-boot.mk software/buildroot/target/device/armadeus/u-boot/u-boot.mk
--- buildroot/target/device/armadeus/u-boot/u-boot.mk	2006-01-27 01:46:08.000000000 +0100
+++ software/buildroot/target/device/armadeus/u-boot/u-boot.mk	2006-01-27 01:39:10.000000000 +0100
@@ -17,7 +17,7 @@ $(DL_DIR)/$(U-BOOT_SOURCE):
 u-boot-source: $(DL_DIR)/$(U-BOOT_SOURCE)
 
 $(U-BOOT_DIR)/.unpacked: $(DL_DIR)/$(U-BOOT_SOURCE)
-	bzcat $(DL_DIR)/$(U-BOOT_SOURCE) | tar -C $(BUILD_DIR) -xvf -
+	bzcat $(DL_DIR)/$(U-BOOT_SOURCE) | tar -C $(BUILD_DIR) $(TAR_OPTIONS) -
 	toolchain/patch-kernel.sh $(U-BOOT_DIR) $(U-BOOT_PACKAGE_DIR) *.patch
 	touch $(U-BOOT_DIR)/.unpacked	
 
@@ -39,11 +39,26 @@ u-boot: $(U-BOOT_DIR)/u-boot.bin $(U-BOO
 
 u-boot-clean:
 	make -C $(U-BOOT_DIR) clean
-	rm $(U-BOOT_DIR)/.configured
+	rm -rf $(U-BOOT_DIR)/tools/mkbrecimage
+	rm -rf $(U-BOOT_DIR)/.configured
 
-u-boot-dirclean:
+u-boot-distclean: u-boot-clean
+	make -C $(U-BOOT_DIR) distclean
+
+u-boot-dirclean: 
 	rm -rf $(U-BOOT_DIR)
 
+u-boot-patch: u-boot-distclean
+	-mkdir -p $(BUILD_DIR)/ref
+	rm -rf $(BUILD_DIR)/ref/u-boot-$(U-BOOT_VER)
+	bzcat $(DL_DIR)/$(U-BOOT_SOURCE) | tar -C $(BUILD_DIR)/ref $(TAR_OPTIONS) -
+	toolchain/patch-kernel.sh $(BUILD_DIR)/ref/u-boot-$(U-BOOT_VER) $(U-BOOT_PACKAGE_DIR) *.patch
+	touch $(BUILD_DIR)/ref/u-boot-$(U-BOOT_VER)/.unpacked
+	-(cd $(BUILD_DIR); \
+	diff -purN -x '*~' ref/u-boot-$(U-BOOT_VER) u-boot-$(U-BOOT_VER) > $(BUILD_DIR)/../newu-boot.diff;\
+	echo -e "\n\nu-boot patch generated!\n")
+
+
 #############################################################
 #
 # Toplevel Makefile options
