diff -purN mkfs-ubifs-HEAD/mkfs.ubifs.h mkfs-ubifs-HEAD.new/mkfs.ubifs.h
--- mtd-utils/mkfs.ubifs/mkfs.ubifs.h	2008-08-12 10:34:49.000000000 +0200
+++ mtd-utils.new/mkfs.ubifs/mkfs.ubifs.h	2009-06-07 11:53:18.000000000 +0200
@@ -35,6 +35,7 @@
 #include <byteswap.h>
 #include <linux/types.h>
 #include <linux/fs.h>
+#include <linux/ext2_fs.h>
 
 #include <getopt.h>
 #include <sys/types.h>
@@ -70,6 +71,31 @@
 #error MKFS_UBIFS_COMPR_ZLIB != UBIFS_COMPR_ZLIB
 #endif
 
+#ifndef FS_COMPR_FL
+#define FS_COMPR_FL EXT2_COMPR_FL
+#endif
+
+#ifndef FS_SYNC_FL
+#define FS_SYNC_FL EXT2_SYNC_FL
+#endif
+
+#ifndef FS_IMMUTABLE_FL
+#define FS_IMMUTABLE_FL EXT2_IMMUTABLE_FL
+#endif
+
+#ifndef FS_APPEND_FL
+#define FS_APPEND_FL EXT2_APPEND_FL
+#endif
+
+#ifndef FS_DIRSYNC_FL
+#define FS_DIRSYNC_FL EXT2_DIRSYNC_FL
+#endif
+
+#ifndef FS_IOC_GETFLAGS
+#define FS_IOC_GETFLAGS EXT2_IOC_GETFLAGS
+#endif
+
+
 extern int verbose;
 extern int debug_level;
 
