--- slideshow/slideshow.cpp	2003-12-08 22:17:38.000000000 +0100
+++ slideshow.mod/slideshow.cpp	2007-09-08 19:05:10.000000000 +0200
@@ -196,43 +196,48 @@
 			SDL_FillRect(m_sdl, &rc, SDL_MapRGB(m_sdl->format, 0, 255, 0));
 			SDL_UpdateRects(m_sdl, 1, &rc);
 		}
-
-		user_command cmd = wait_for_command();
-
+
+		user_command cmd = wait_for_command();
+
 		switch(cmd)
-		{
-		case c_quit:
-			stop_timer();
-			return true;
-		case c_first_slide:
+		{
+		case c_quit:
+			stop_timer();
+			return true;
+		case c_first_slide:
 			stop_timer();
 			m_image_index = 0;
 			break;
 		case c_last_slide:
-			stop_timer();
+			stop_timer();
 			m_image_index = m_file_list->count() - 1;
-			break;
+			break;
 		case c_prev_slide:
-			stop_timer();
+			stop_timer();
 			if ( m_image_index > 0 )
 				m_image_index--;
 			break;
 		case c_next_slide:
-			stop_timer();
-			// fall through
-		case c_timer_advance:
-			if ( m_image_index < m_file_list->count() - 1 )
+			stop_timer();
+			// fall through
+		case c_timer_advance:
+			if ( m_image_index <= m_file_list->count() - 1 )
 				m_image_index++;
-			break;
-		case c_redraw:
 			break;
-		}
+		case c_redraw:
+			break;
+		}
 
 		// stop the auto-advance when the last picture is about to be shown
-		if ( m_image_index == m_file_list->count() - 1 )
-			stop_timer();
-	}
-
+		if ( m_image_index == m_file_list->count() /*- 1*/ )
+		{
+			//stop_timer();
+			m_image_index = 0;
+			m_prev_image_index = -1;
+		}
+
+	}
+
 	return true;
 }
 
