--- slideshow/Makefile	2003-11-29 21:08:54.000000000 +0100
+++ slideshow.mod/Makefile	2007-09-05 12:08:56.000000000 +0200
@@ -1,6 +1,12 @@
+ifeq ($(ARCH), arm)
+CFLAGS = -O2 `sh ../SDL-1.2.11/sdl-config --cflags`
+LFLAGS = `sh ../SDL-1.2.11/sdl-config --libs` -lSDL_image
+CC = arm-linux-g++
+else
 CFLAGS = -O2 -march=i586 `sdl-config --cflags`
 LFLAGS = `sdl-config --libs` -lSDL_image
 CC = g++
+endif
 EXE = slideshow
 OBJ = file_list.o fontdata.o main.o SDLFont.o slideshow.o
 
--- slideshow/SDLFont.h	2003-11-29 21:08:54.000000000 +0100
+++ slideshow.mod/SDLFont.h	2007-09-05 12:11:17.000000000 +0200
@@ -34,7 +34,7 @@
 // The value of the top-left pixel in the font bitmap will be
 //  considered transparent.  Anything else is opaque.
 
-#include <SDL/SDL.h>
+#include "SDL.h"
 
 class SDLFont
 {
