Index: linux-2.6.29/arch/arm/mach-imx/include/mach/imxfb.h
===================================================================
--- linux-2.6.29.orig/arch/arm/mach-imx/include/mach/imxfb.h	2009-03-30 10:52:07.000000000 +0200
+++ linux-2.6.29/arch/arm/mach-imx/include/mach/imxfb.h	2009-03-30 10:52:38.000000000 +0200
@@ -14,6 +14,7 @@
 #define PCR_BPIX_8	(3 << 25)
 #define PCR_BPIX_12	(4 << 25)
 #define PCR_BPIX_16	(4 << 25)
+#define PCR_BPIX_MASK   (7 << 25)
 #define PCR_PIXPOL	(1 << 24)
 #define PCR_FLMPOL	(1 << 23)
 #define PCR_LPPOL	(1 << 22)
