Index: linux-2.6.29.4/drivers/video/Kconfig
===================================================================
--- linux-2.6.29.4.orig/drivers/video/Kconfig	2009-03-24 00:12:14.000000000 +0100
+++ linux-2.6.29.4/drivers/video/Kconfig	2009-06-04 10:03:43.000000000 +0200
@@ -395,6 +395,7 @@
 	  If you plan to use the LCD display with your SA-1100 system, say
 	  Y here.
 
+
 config FB_IMX
 	tristate "Motorola i.MX LCD support"
 	depends on FB && ARM && ARCH_IMX
@@ -402,6 +403,42 @@
 	select FB_CFB_COPYAREA
 	select FB_CFB_IMAGEBLIT
 
+choice
+        depends on FB_IMX || FB_MXC
+        prompt "LCD Panel"
+        default FB_IMX_MICROTIPS_MTF_T035_TFT
+
+config FB_IMX_SHARP_LQ043_TFT
+        bool "Sharp LQ043"
+        help
+          Sharp LQ043, a 4.3" 24 bits color 480x272, TFT panel.
+          The native resolution is 480x272.
+
+config FB_IMX_MICROTIPS_MTF_T035_TFT
+        bool "Microtips MTF-T035"
+        help
+          Microtips MTF-T035, a 3.5" 24 bits color 320x240, TFT panel.
+          The native resolution is 320x240.
+
+config FB_IMX_SHARP_LQ057_TFT
+        bool "Sharp LQ057"
+        help
+          Sharp LQ057, a 3.5" 18 bits color QVGA, TFT panel.
+          The native resolution is 320x240.
+
+config FB_IMX_OPTREX_F51900_CSTN
+        bool "Optrex F-51900"
+        help
+          Optrex 5.7" 8 bits color STN.
+	   The native resolution is 320x240.
+
+config FB_IMX_MOTOROLA_A910_TFT
+        bool "Motorola A910 TFT"
+        help
+          Motorola A910 screen, a 1.2" color QVGA, TFT panel.
+          The native resolution is 240x320.
+endchoice
+
 config FB_CYBER2000
 	tristate "CyberPro 2000/2010/5000 support"
 	depends on FB && PCI && (BROKEN || !SPARC64)
