--- linux-org/arch/arm/mach-imx/include/mach/hardware.h
+++ linux/arch/arm/mach-imx/include/mach/hardware.h
@@ -88,4 +88,8 @@
 #include "mx1ads.h"
 #endif
 
+#ifdef CONFIG_MACH_APF9328
+#include "apf9328.h"
+#endif
+
 #endif
--- linux-org/arch/arm/mach-imx/Kconfig
+++ linux/arch/arm/mach-imx/Kconfig
@@ -1,6 +1,10 @@
 menu "IMX Implementations"
 	depends on ARCH_IMX
-
+
+choice
+    prompt "Development board selection"
+    depends on ARCH_IMX
+
 config ARCH_MX1ADS
 	bool "mx1ads"
 	depends on ARCH_IMX
@@ -12,4 +16,38 @@
 	help
 	  Say Y here if you are using the Motorola MX1ADS board
 
+config MACH_APF9328
+    bool "apf9328"
+    depends on ARCH_IMX
+    help
+      Say Yes here if you are using the Armadeus APF9328 development board
+
+endchoice
+
+choice
+       prompt "APF9328 Flash available"
+       depends on MACH_APF9328
+       default IMX_APF9328_FLASH_16MB
+
+config IMX_APF9328_FLASH_8MB
+       bool "8MB"
+
+config IMX_APF9328_FLASH_16MB
+       bool "16MB"
+
+endchoice
+
+choice
+       prompt "APF9328 RAM available"
+       depends on MACH_APF9328
+       default IMX_APF9328_RAM_32MB
+
+config IMX_SCB9328_RAM_16MB
+       bool "16MB"
+
+config IMX_SCB9328_RAM_32MB
+       bool "32MB"
+
+endchoice
+
 endmenu
--- linux-org/arch/arm/mach-imx/Makefile
+++ linux/arch/arm/mach-imx/Makefile
@@ -10,6 +10,7 @@
 
 # Specific board support
 obj-$(CONFIG_ARCH_MX1ADS) += mx1ads.o
+obj-$(CONFIG_MACH_APF9328) += apf9328.o
 
 # Support for blinky lights
 led-y := leds.o
--- linux-org/arch/arm/mach-imx/Makefile.boot
+++ linux/arch/arm/mach-imx/Makefile.boot
@@ -1,1 +1,2 @@
     zreladdr-$(CONFIG_ARCH_MX1ADS)	:= 0x08008000
+    zreladdr-$(CONFIG_MACH_APF9328) := 0x08008000
\ No newline at end of file
