Index: linux-2.6.29/drivers/usb/gadget/Kconfig
===================================================================
--- linux-2.6.29.orig/drivers/usb/gadget/Kconfig	2009-03-24 00:12:14.000000000 +0100
+++ linux-2.6.29/drivers/usb/gadget/Kconfig	2009-03-30 10:55:52.000000000 +0200
@@ -439,6 +439,27 @@
 	default USB_GADGET
 	select USB_GADGET_SELECTED
 
+config USB_GADGET_IMX
+	boolean "i.MX: MX1, MXL"
+	depends on ARCH_IMX
+	help
+	   Freescale's i.MX series include an integrated full speed 
+	   USB 1.1 device controller.  The controllers in the i.MX series
+	   are register-compatible.
+
+	   It has Six fixed-function endpoints, as well as endpoint
+	   zero (for control transfers).
+
+	   Say "y" to link the driver statically, or "m" to build a
+	   dynamically linked module called "imx_udc" and force all
+	   gadget drivers to also be dynamically linked.
+
+config USB_IMX
+	tristate
+	depends on USB_GADGET_IMX
+	default USB_GADGET
+   select USB_GADGET_SELECTED
+
 config USB_GADGET_GOKU
 	boolean "Toshiba TC86C001 'Goku-S'"
 	depends on PCI
