--- linux-org/arch/arm/mach-imx/include/mach/imxfb.h
+++ linux/arch/arm/mach-imx/include/mach/imxfb.h
@@ -35,3 +35,12 @@
 	void (*backlight_power)(int);
 };
 void set_imx_fb_info(struct imxfb_mach_info *hard_imx_fb_info);
+
+// --- Backlight handling ---
+struct imxbl_machinfo {
+	int max_intensity;
+	int default_intensity;
+	int limit_mask; // When battery is low
+	void (*set_bl_intensity)(int intensity);
+};
+extern void imxbl_limit_intensity(int limit);
