Index: linux-2.6.29/arch/arm/mach-imx/generic.c
===================================================================
--- linux-2.6.29.orig/arch/arm/mach-imx/generic.c	2009-03-30 11:16:06.000000000 +0200
+++ linux-2.6.29/arch/arm/mach-imx/generic.c	2009-03-30 11:18:12.000000000 +0200
@@ -250,7 +250,7 @@
 
 static struct imx_fb_platform_data imx_fb_info;
 
-void __init set_imx_fb_info(struct imx_fb_platform_data *hard_imx_fb_info)
+void set_imx_fb_info(struct imx_fb_platform_data *hard_imx_fb_info)
 {
 	memcpy(&imx_fb_info,hard_imx_fb_info,sizeof(struct imx_fb_platform_data));
 }
