--- linux-org/arch/arm/mach-imx/include/mach/imx-regs.h	2008-07-18 18:51:40.000000000 +0200
+++ linux/arch/arm/mach-imx/include/mach/imx-regs.h	2008-07-18 09:55:17.000000000 +0200
@@ -316,11 +316,12 @@
 #define RTOR(x) __REG2( IMX_DMAC_BASE + 0x98, (x) << 6)	/* Request timeout Registers */
 #define BUCR(x) __REG2( IMX_DMAC_BASE + 0x98, (x) << 6)	/* Bus Utilization Registers */
 
+#define MAX_DMA_CNTR	   (0xffffff)
 #define DCR_DRST           (1<<1)
 #define DCR_DEN            (1<<0)
 #define DBTOCR_EN          (1<<15)
 #define DBTOCR_CNT(x)      ((x) & 0x7fff )
-#define CNTR_CNT(x)        ((x) & 0xffffff )
+#define CNTR_CNT(x)        ((x) & MAX_DMA_CNTR )
 #define CCR_DMOD_LINEAR    ( 0x0 << 12 )
 #define CCR_DMOD_2D        ( 0x1 << 12 )
 #define CCR_DMOD_FIFO      ( 0x2 << 12 )
