--- linux-2.6.27.2.original/arch/arm/mach-imx/clock.c	2009-01-14 10:20:49.000000000 +0100
+++ linux-2.6.27.2/arch/arm/mach-imx/clock.c	2009-01-14 09:55:02.000000000 +0100
@@ -151,6 +151,11 @@
 	.get_rate = imx_get_perclk3,
 };
 
+static struct clk clk_ssi_clk = {
+	.name = "ssi_clk",
+	.get_rate = imx_get_perclk3,
+};
+
 static struct clk clk_usb_clk = {
 	.name = "usb_clk",
 	.get_rate = imx_get_usb_clk,
@@ -165,6 +170,7 @@
 	&clk_mcu_clk,
 	&clk_uart_clk,
 	&clk_usb_clk,
+	&clk_ssi_clk,
 };
 
 static LIST_HEAD(clocks);
