Index: linux-2.6.29/drivers/video/mxc/mxcfb_modedb.c
===================================================================
--- linux-2.6.29.orig/drivers/video/mxc/mxcfb_modedb.c	2009-03-31 10:57:25.000000000 +0200
+++ linux-2.6.29/drivers/video/mxc/mxcfb_modedb.c	2009-03-31 10:57:32.000000000 +0200
@@ -88,6 +88,19 @@
 		.vmode = FB_VMODE_NONINTERLACED,
 		.flag = 0,
 		},
+        [6] = {
+		.name = "Sharp-LQ057",
+		.refresh = 60,
+		.xres = 320,
+		.yres = 240,
+		.pixclock = 111111, /* picoS */
+		.left_margin = 52, .right_margin = 28, /* at least 3 & 1 */
+		.upper_margin = 2, .lower_margin = 3,
+		.hsync_len = 0, .vsync_len = 7,
+		.sync = FB_SYNC_CLK_IDLE_EN,
+		.vmode = FB_VMODE_NONINTERLACED, /* TO CHECK !! */
+		.flag = 0,
+	},
 };
 
 int mxcfb_modedb_sz = ARRAY_SIZE(mxcfb_modedb);
