--- linux-2.6.27.2.original/arch/arm/plat-mxc/include/mach/imxfb.h	2008-12-19 16:20:18.000000000 +0100
+++ linux-2.6.27.2.mod/arch/arm/plat-mxc/include/mach/imxfb.h	2008-12-19 16:18:44.000000000 +0100
@@ -83,3 +83,13 @@
 };
 
 void set_imx_fb_info(struct imx_fb_platform_data *);
+
+/* --- Backlight handling --- */
+struct imxbl_machinfo {
+	int max_intensity;
+	int default_intensity;
+	int limit_mask; // When battery is low
+	void (*set_bl_intensity)(int intensity);
+};
+extern void imxbl_limit_intensity(int limit);
+
