Index: linux-2.6.29.6/arch/arm/plat-mxc/gpio.c
===================================================================
--- linux-2.6.29.6.orig/arch/arm/plat-mxc/gpio.c	2010-01-05 15:27:43.000000000 +0100
+++ linux-2.6.29.6/arch/arm/plat-mxc/gpio.c	2010-01-05 15:28:24.000000000 +0100
@@ -226,7 +226,7 @@
 	void __iomem *reg = port->base + GPIO_DR;
 	u32 l;
 
-	l = (__raw_readl(reg) & (~(1 << offset))) | (value << offset);
+	l = (__raw_readl(reg) & (~(1 << offset))) | ((value?1:0) << offset);
 	__raw_writel(l, reg);
 }
 
