--- linux-org/arch/arm/mach-imx/irq.c	2006-09-20 05:42:06.000000000 +0200
+++ linux-imx/arch/arm/mach-imx/irq.c	2006-12-09 17:37:53.386928000 +0100
@@ -69,6 +69,12 @@ imx_unmask_irq(unsigned int irq)
 }
 
 static int
+imx_irq_type(unsigned int _irq, unsigned int type)
+{
+    return 0;
+}
+
+static int
 imx_gpio_irq_type(unsigned int _irq, unsigned int type)
 {
 	unsigned int irq_type = 0, irq, reg, bit;
@@ -205,6 +211,7 @@ static struct irq_chip imx_internal_chip
 	.ack = imx_mask_irq,
 	.mask = imx_mask_irq,
 	.unmask = imx_unmask_irq,
+	.set_type = imx_irq_type,
 };
 
 static struct irq_chip imx_gpio_chip = {
 
