Index: linux-2.6.27.13/arch/arm/plat-mxc/include/mach/debug-macro.S
===================================================================
--- linux-2.6.27.13.orig/arch/arm/plat-mxc/include/mach/debug-macro.S	2008-10-10 00:13:53.000000000 +0200
+++ linux-2.6.27.13/arch/arm/plat-mxc/include/mach/debug-macro.S	2009-04-14 13:56:06.000000000 +0200
@@ -28,6 +28,9 @@
 #ifdef CONFIG_MACH_PCM038
 #include <mach/board-pcm038.h>
 #endif
+#ifdef CONFIG_MACH_APF27
+#include <mach/board-apf27.h>
+#endif
 		.macro	addruart,rx
 		mrc	p15, 0, \rx, c1, c0
 		tst	\rx, #1			@ MMU enabled?
