--- linux-2.6.27.2.ref/fs/jffs2/scan.c	2008-12-19 13:47:54.000000000 +0100
+++ linux-2.6.27.2/fs/jffs2/scan.c	2008-12-19 13:51:44.000000000 +0100
@@ -647,8 +647,8 @@ scan_more:
 			inbuf_ofs = ofs - buf_ofs;
 			while (inbuf_ofs < scan_end) {
 				if (unlikely(*(uint32_t *)(&buf[inbuf_ofs]) != 0xffffffff)) {
-					printk(KERN_WARNING "Empty flash at 0x%08x ends at 0x%08x\n",
-					       empty_start, ofs);
+					D1(printk(KERN_WARNING "Empty flash at 0x%08x ends at 0x%08x\n",
+					       empty_start, ofs));
 					if ((err = jffs2_scan_dirty_space(c, jeb, ofs-empty_start)))
 						return err;
 					goto scan_more;
@@ -965,9 +965,9 @@ static int jffs2_scan_inode_node(struct 
 	/* Check the node CRC in any case. */
 	crc = crc32(0, ri, sizeof(*ri)-8);
 	if (crc != je32_to_cpu(ri->node_crc)) {
-		printk(KERN_NOTICE "jffs2_scan_inode_node(): CRC failed on "
+		D1(printk(KERN_NOTICE "jffs2_scan_inode_node(): CRC failed on "
 		       "node at 0x%08x: Read 0x%08x, calculated 0x%08x\n",
-		       ofs, je32_to_cpu(ri->node_crc), crc);
+		       ofs, je32_to_cpu(ri->node_crc), crc));
 		/*
 		 * We believe totlen because the CRC on the node
 		 * _header_ was OK, just the node itself failed.
--- linux-2.6.27.2.ref/fs/jffs2/readinode.c	2008-12-19 14:03:21.000000000 +0100
+++ linux-2.6.27.2/fs/jffs2/readinode.c	2008-12-19 14:03:44.000000000 +0100
@@ -105,7 +105,7 @@ static int check_node_data(struct jffs2_
 #endif
 
 	if (crc != tn->data_crc) {
-		JFFS2_NOTICE("wrong data CRC in data node at 0x%08x: read %#08x, calculated %#08x.\n",
+		dbg_readinode("wrong data CRC in data node at 0x%08x: read %#08x, calculated %#08x.\n",
 			     ref_offset(ref), tn->data_crc, crc);
 		return 1;
 	}
