Increases max DMA zone size for MXC platform.

Signed-off-by: Nicolas Colombain <nicolas.colombain@armadeus.com>
Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: linux-2.6.27.13/arch/arm/plat-mxc/include/mach/memory.h
===================================================================
--- linux-2.6.27.13.orig/arch/arm/plat-mxc/include/mach/memory.h	2009-03-27 17:52:54.000000000 +0100
+++ linux-2.6.27.13/arch/arm/plat-mxc/include/mach/memory.h	2009-03-27 17:58:35.000000000 +0100
@@ -26,4 +26,7 @@
  */
 #define __bus_to_virt(a)	__phys_to_virt(a)
 
+/* Override the ARM default */
+#define CONSISTENT_DMA_SIZE SZ_4M
+
 #endif /* __ASM_ARCH_MXC_MEMORY_H__ */
