Needed for POD IRQ handler registering (module).

--- linux-org/arch/arm/kernel/irq.c	2007-10-12 18:43:44.000000000 +0200
+++ linux/arch/arm/kernel/irq.c	2008-07-24 16:43:52.000000000 +0200
@@ -153,6 +153,7 @@
 		desc->status &= ~IRQ_NOAUTOEN;
 	spin_unlock_irqrestore(&desc->lock, flags);
 }
+EXPORT_SYMBOL(set_irq_flags);
 
 void __init init_IRQ(void)
 {
--- linux-org/arch/arm/mach-imx/include/mach/irqs.h	2008-07-29 17:26:13.000000000 +0200
+++ linux/arch/arm/mach-imx/include/mach/irqs.h	2008-07-24 17:26:17.000000000 +0200
@@ -112,6 +112,10 @@
 /* decode irq number to use with IMR(x), ISR(x) and friends */
 #define IRQ_TO_REG(irq) ((irq - IMX_IRQS) >> 5)
 
-#define NR_IRQS (IRQ_GPIOD(32) + 1)
+#define IRQ_FPGA_START (IRQ_GPIOD(32) + 1)
+#define IRQ_FPGA(x) (IRQ_FPGA_START + x)
+
+//#define NR_IRQS (IRQ_GPIOD(32) + 1)
+#define NR_IRQS (IRQ_FPGA(16) + 1)
 #define IRQ_GPIO(x)
 #endif
--- linux-org/kernel/irq/chip.c	2007-10-12 18:43:44.000000000 +0200
+++ linux/kernel/irq/chip.c	2008-07-24 16:47:35.000000000 +0200
@@ -511,6 +511,7 @@
 out_unlock:
 	spin_unlock(&desc->lock);
 }
+EXPORT_SYMBOL(handle_edge_irq);
 
 /**
  *	handle_percpu_IRQ - Per CPU local irq handler
@@ -587,6 +588,7 @@
 	}
 	spin_unlock_irqrestore(&desc->lock, flags);
 }
+EXPORT_SYMBOL(__set_irq_handler);
 
 void
 set_irq_chip_and_handler(unsigned int irq, struct irq_chip *chip,
@@ -595,6 +597,7 @@
 	set_irq_chip(irq, chip);
 	__set_irq_handler(irq, handle, 0, NULL);
 }
+EXPORT_SYMBOL(set_irq_chip_and_handler);
 
 void
 set_irq_chip_and_handler_name(unsigned int irq, struct irq_chip *chip,
--- linux-org/kernel/irq/handle.c	2007-10-12 18:43:44.000000000 +0200
+++ linux/kernel/irq/handle.c	2008-07-24 17:02:07.000000000 +0200
@@ -59,6 +59,7 @@
 #endif
 	}
 };
+EXPORT_SYMBOL(irq_desc);
 
 /*
  * What should we do if we get a hw irq event on an illegal vector?
