--- linux-org/arch/arm/mach-imx/generic.c	2006-10-20 00:13:49.000000000 +0200
+++ linux-imx/arch/arm/mach-imx/generic.c	2006-10-18 22:47:19.000000000 +0200
@@ -229,10 +229,13 @@ static struct resource imx_mmc_resources
 
 static u64 imxmmmc_dmamask = 0xffffffffUL;
 
+static struct imxmmc_platform_data imx_mmc_info;
+
 static struct platform_device imx_mmc_device = {
 	.name		= "imx-mmc",
 	.id		= 0,
 	.dev		= {
+		.platform_data	= &imx_mmc_info,
 		.dma_mask = &imxmmmc_dmamask,
 		.coherent_dma_mask = 0xffffffff,
 	},
@@ -242,7 +245,7 @@ static struct platform_device imx_mmc_de
 
 void __init imx_set_mmc_info(struct imxmmc_platform_data *info)
 {
-	imx_mmc_device.dev.platform_data = info;
+	memcpy(&imx_mmc_info,info,sizeof(struct imxmmc_platform_data));
 }
 EXPORT_SYMBOL(imx_set_mmc_info);
 
