--- linux-2.6.27.2/drivers/video/mxc/mx2fb.c	2008-12-02 15:49:06.000000000 +0100
+++ linux-2.6.27.2.mod/drivers/video/mxc/mx2fb.c	2008-12-03 17:31:04.000000000 +0100
@@ -30,7 +30,7 @@
 #include <linux/interrupt.h>
 #include <linux/clk.h>
 #include <asm/uaccess.h>
-#include <asm/arch/imx_fb.h>
+#include <mach/imx_fb.h>
 
 #include "mx2fb.h"
 
@@ -997,7 +997,7 @@
 	/* Read to clear the status */
 	status = __raw_readl(LCDC_REG(LCDC_LISR));
 
-	if (request_irq(INT_LCDC, mx2fb_isr, 0, "LCDC", 0))	/* FIXME source of interrupt number */
+	if (request_irq(MXC_INT_LCDC, mx2fb_isr, 0, "LCDC", 0))	/* FIXME source of interrupt number */
 		pr_info("Request LCDC IRQ failed.\n");
 	else {
 		spin_lock_irqsave(&mx2fb_notifier_list.lock, flags);
@@ -1028,7 +1028,7 @@
 	/* Disable all LCDC interrupt */
 	__raw_writel(0x0, LCDC_REG(LCDC_LIER));
 
-	free_irq(INT_LCDC, 0);
+	free_irq(MXC_INT_LCDC, 0);
 }
 
 int mx2fb_register_client(struct notifier_block *nb)
--- linux-2.6.27.2/drivers/video/mxc/mxcfb_modedb.c	2008-12-02 15:49:07.000000000 +0100
+++ linux-2.6.27.2.mod/drivers/video/mxc/mxcfb_modedb.c	2008-12-03 17:31:04.000000000 +0100
@@ -17,7 +17,7 @@
  */
 
 #include <linux/kernel.h>
-#include <asm/arch/imx_fb.h>
+#include <mach/imx_fb.h>
 
 const struct fb_videomode mxcfb_modedb[] = {
 	[0] = {
