--- linux-org/drivers/usb/gadget/Kconfig	2006-09-20 05:42:06.000000000 +0200
+++ linux-imx/drivers/usb/gadget/Kconfig	2006-10-01 14:30:04.000000000 +0200
@@ -251,6 +251,27 @@ config USB_PXA27X
 	default USB_GADGET
 	select USB_GADGET_SELECTED
 
+config USB_GADGET_IMX
+	boolean "i.MX: MX1, MXL"
+	depends on ARCH_IMX
+	help
+	   Freescale's i.MX series include an integrated full speed 
+	   USB 1.1 device controller.  The controllers in the i.MX series
+	   are register-compatible.
+
+	   It has Six fixed-function endpoints, as well as endpoint
+	   zero (for control transfers).
+
+	   Say "y" to link the driver statically, or "m" to build a
+	   dynamically linked module called "imx_udc" and force all
+	   gadget drivers to also be dynamically linked.
+
+config USB_IMX
+	tristate
+	depends on USB_GADGET_IMX
+	default USB_GADGET
+   select USB_GADGET_SELECTED
+
 config USB_GADGET_GOKU
 	boolean "Toshiba TC86C001 'Goku-S'"
 	depends on PCI
--- linux-org/drivers/usb/gadget/Makefile	2006-09-20 05:42:06.000000000 +0200
+++ linux-imx/drivers/usb/gadget/Makefile	2006-10-01 14:30:04.000000000 +0200
@@ -10,6 +10,7 @@
 obj-$(CONFIG_USB_AMD5536UDC)	+= amd5536udc.o
 obj-$(CONFIG_USB_PXA25X)	+= pxa25x_udc.o
 obj-$(CONFIG_USB_PXA27X)	+= pxa27x_udc.o
+obj-$(CONFIG_USB_IMX)		+= imx_udc.o
 obj-$(CONFIG_USB_GOKU)		+= goku_udc.o
 obj-$(CONFIG_USB_OMAP)		+= omap_udc.o
 obj-$(CONFIG_USB_LH7A40X)	+= lh7a40x_udc.o
