--- ref/linux-2.6.27/drivers/i2c/busses/Kconfig	2006-10-14 05:34:03.000000000 +0200
+++ linux-2.6.27/drivers/i2c/busses/Kconfig	2007-01-14 00:35:15.000000000 +0100
@@ -612,6 +612,12 @@ config I2C_PCA_ISA
 
 	  This driver can also be built as a module.  If so, the module
 	  will be called i2c-pca-isa.
+
+config I2C_IMX
+	tristate "i2c bus on i.MX"
+	depends on I2C 
+	help
+	  i2c implementation on i.MX Freescale processors
 
 	  This device is almost undetectable and using this driver on a
 	  system which doesn't have this device will result in long
--- ref/linux-2.6.27/drivers/i2c/busses/Makefile	2006-10-14 05:34:03.000000000 +0200
+++ linux-2.6.27/drivers/i2c/busses/Makefile	2007-01-14 00:35:15.000000000 +0100
@@ -43,6 +43,7 @@ # Embebbed system I2C/SMBus host controller drivers
 obj-$(CONFIG_I2C_PXA)		+= i2c-pxa.o
 obj-$(CONFIG_I2C_S3C2410)	+= i2c-s3c2410.o
 obj-$(CONFIG_I2C_SH7760)	+= i2c-sh7760.o
+obj-$(CONFIG_I2C_IMX)		+= i2c-imx.o
 obj-$(CONFIG_I2C_SH_MOBILE)	+= i2c-sh_mobile.o
 obj-$(CONFIG_I2C_SIMTEC)	+= i2c-simtec.o
 obj-$(CONFIG_I2C_VERSATILE)	+= i2c-versatile.o
