Specific for Armadeus (needed by quiltify.sh script).

Index: buildroot/boot/u-boot/u-boot.mk
===================================================================
--- buildroot.orig/boot/u-boot/u-boot.mk	2010-10-20 13:42:14.000000000 +0200
+++ buildroot/boot/u-boot/u-boot.mk	2010-10-20 13:42:17.000000000 +0200
@@ -82,6 +82,8 @@
 		| tar $(TAR_STRIP_COMPONENTS)=1 -C $(@D) $(TAR_OPTIONS) -
 	touch $@
 
+u-boot-src-unpacked: $(U_BOOT_DIR)/.unpacked
+
 $(U_BOOT_DIR)/.patched: $(U_BOOT_DIR)/.unpacked
 	toolchain/patch-kernel.sh $(U_BOOT_DIR) boot/u-boot \
 		u-boot-$(U_BOOT_VERSION)-\*.patch \
