Index: buildroot/linux/linux.mk
===================================================================
--- buildroot.orig/linux/linux.mk	2011-01-27 15:24:26.000000000 +0100
+++ buildroot/linux/linux.mk	2011-01-27 15:24:32.000000000 +0100
@@ -113,6 +113,8 @@
 	$(call KCONFIG_ENABLE_OPT,CONFIG_INITRAMFS_COMPRESSION_GZIP,$(@D)/.config)
 endif
 ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
+	$(SED) "s,^ARCH.*,ARCH=$(KERNEL_ARCH),g;" $(LINUX26_DIR)/Makefile
+	$(SED) "s,^CROSS_COMPILE.*,CROSS_COMPILE=$(KERNEL_CROSS),g;" $(LINUX26_DIR)/Makefile
 	(ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ $(LINUX26_DIR)/drivers/armadeus)
 endif
 	$(TARGET_MAKE_ENV) $(MAKE) $(LINUX26_MAKE_FLAGS) -C $(@D) oldconfig
