--- buildroot/package/Config.in.ref     2008-11-03 15:40:51.000000000 +0100
+++ buildroot/package/Config.in 2008-11-03 15:41:51.000000000 +0100
@@ -249,6 +249,7 @@ source "package/e2fsprogs/Config.in"
 source "package/eeprog/Config.in"
 source "package/fconfig/Config.in"
 source "package/fis/Config.in"
+source "package/fpgaregs/Config.in"
 source "package/fuse/Config.in"
 source "package/gadgetfs-test/Config.in"
 source "package/hal/Config.in"
