Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-01-25 10:04:00.000000000 +0100
+++ buildroot/package/Config.in	2010-01-25 10:04:02.000000000 +0100
@@ -311,6 +311,7 @@
 source "package/dialog/Config.in"
 
 comment "graphic libraries"
+source "package/allegro/Config.in"
 source "package/directfb/Config.in"
 source "package/directfb-examples/Config.in"
 source "package/fbdump/Config.in"
