Index: buildroot/target/jffs2/jffs2root.mk
===================================================================
--- buildroot.orig/target/jffs2/jffs2root.mk	2010-01-25 10:03:35.000000000 +0100
+++ buildroot/target/jffs2/jffs2root.mk	2010-01-25 10:04:15.000000000 +0100
@@ -58,7 +58,10 @@
 ifneq ($(BR2_HAVE_INFOPAGES),y)
 	@rm -rf $(TARGET_DIR)/usr/info
 endif
-	@rmdir -p --ignore-fail-on-non-empty $(TARGET_DIR)/usr/share
+	@test -d $(TARGET_DIR)/usr/share && \
+		rmdir -p --ignore-fail-on-non-empty $(TARGET_DIR)/usr/share || \
+		true
+#	@rmdir -p --ignore-fail-on-non-empty $(TARGET_DIR)/usr/share
 	$(if $(TARGET_LDCONFIG),test -x $(TARGET_LDCONFIG) && $(TARGET_LDCONFIG) -r $(TARGET_DIR) 2>/dev/null)
 	# Use fakeroot to pretend all target binaries are owned by root
 	rm -f $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(JFFS2_TARGET))
