Add fbgrab package to BR.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-19 17:34:21.000000000 +0200
+++ buildroot/package/Config.in	2010-06-19 17:38:26.000000000 +0200
@@ -338,6 +338,7 @@
 comment "--> May be broken in busybox"
 source "package/fbv/Config.in"
 source "package/fbset/Config.in"
+source "package/fbgrab/Config.in"
 
 comment "other GUIs"
 source "package/qte/Config.in"
Index: buildroot/package/fbgrab/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/fbgrab/Config.in	2010-06-19 17:38:39.000000000 +0200
@@ -0,0 +1,9 @@
+config BR2_PACKAGE_FBGRAB
+	bool "fbgrab"
+	select BR2_PACKAGE_LIBPNG
+	select BR2_PACKAGE_ZLIB
+	help
+	  FBGrab is a framebuffer screenshot program, capturing the Linux
+	  framebuffer and converting it to a png-picture.
+
+	  http://hem.bredband.net/gmogmo/fbgrab/
Index: buildroot/package/fbgrab/fbgrab.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/fbgrab/fbgrab.mk	2010-06-19 17:34:21.000000000 +0200
@@ -0,0 +1,48 @@
+#############################################################
+#
+# FBGRAB
+#
+#############################################################
+FBGRAB_VERSION:=1.0
+FBGRAB_SOURCE:=fbgrab-$(FBGRAB_VERSION).tar.gz
+FBGRAB_SITE:=http://hem.bredband.net/gmogmo/fbgrab/
+FBGRAB_DIR:=$(BUILD_DIR)/fbgrab-$(FBGRAB_VERSION)
+FBGRAB_CAT:=$(ZCAT)
+FBGRAB_BINARY:=fbgrab
+FBGRAB_TARGET_BINARY:=usr/bin/$(FBGRAB_BINARY)
+
+$(DL_DIR)/$(FBGRAB_SOURCE):
+	$(call DOWNLOAD,$(FBGRAB_SITE),$(FBGRAB_SOURCE))
+
+$(FBGRAB_DIR)/.unpacked: $(DL_DIR)/$(FBGRAB_SOURCE)
+	$(FBGRAB_CAT) $(DL_DIR)/$(FBGRAB_SOURCE) | tar -C $(BUILD_DIR) $(TAR_OPTIONS) -
+	toolchain/patch-kernel.sh $(FBGRAB_DIR) package/fbgrab/ fbgrab-$(FBGRAB_VERSION)\*.patch\*
+	touch $@
+
+$(FBGRAB_DIR)/$(FBGRAB_BINARY): $(FBGRAB_DIR)/.unpacked
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(FBGRAB_DIR)
+	touch -c $@
+
+$(TARGET_DIR)/$(FBGRAB_TARGET_BINARY): $(FBGRAB_DIR)/$(FBGRAB_BINARY)
+	$(INSTALL) -m 755 $(FBGRAB_DIR)/$(FBGRAB_BINARY) $(TARGET_DIR)/$(FBGRAB_TARGET_BINARY)
+	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/$(FBGRAB_TARGET_BINARY)
+	touch -c $@
+
+fbgrab: libpng zlib $(TARGET_DIR)/$(FBGRAB_TARGET_BINARY)
+
+fbgrab-clean:
+	rm -f $(TARGET_DIR)/$(FBGRAB_TARGET_BINARY)
+	-$(MAKE) -C $(FBGRAB_DIR) clean
+
+fbgrab-dirclean:
+	rm -rf $(FBGRAB_DIR)
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_FBGRAB)),y)
+TARGETS+=fbgrab
+endif
+
Index: buildroot/package/fbgrab/fbgrab-1.0-make_it_cross-compile.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/fbgrab/fbgrab-1.0-make_it_cross-compile.patch	2010-06-19 17:34:21.000000000 +0200
@@ -0,0 +1,14 @@
+diff -urN fbgrab-1.0/Makefile fbgrab-1.0.mod/Makefile
+--- fbgrab-1.0/Makefile	2002-03-20 20:49:06.000000000 +0100
++++ fbgrab-1.0.mod/Makefile	2008-06-06 21:42:56.000000000 +0200
+@@ -4,8 +4,8 @@
+ ###
+ 
+ fbgrab: fbgrab.c
+-	splint +posixlib fbgrab.c
+-	gcc -g -Wall fbgrab.c -lpng -lz -o fbgrab
++	#splint +posixlib fbgrab.c
++	$(CC) -Wall fbgrab.c -lpng -lz -o fbgrab
+ 
+ install:
+ 	strip fbgrab
