Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-01-25 10:04:25.000000000 +0100
+++ buildroot/package/Config.in	2010-01-25 10:05:15.000000000 +0100
@@ -338,6 +338,7 @@
 comment "--> May be broken in busybox"
 source "package/fbv/Config.in"
 source "package/fbset/Config.in"
+source "package/fbgrab/Config.in"
 
 comment "other GUIs"
 source "package/qte/Config.in"
