Install PrBoom only once.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Integrated in latest BR.

Index: buildroot/package/games/prboom/prboom.mk
===================================================================
--- buildroot/package/games/prboom/prboom.mk	(revision 23937)
+++ buildroot/package/games/prboom/prboom.mk	(copie de travail)
@@ -42,6 +42,7 @@
 	$(INSTALL) -D $(PRBOOM_DIR)/data/prboom.wad $(TARGET_DIR)/usr/share/games/doom/prboom.wad
 	$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/games/prboom
 	$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/games/prboom-game-server
+	touch $@
 
 $(PRBOOM_TARGET_CLEAN):
 	rm -rf $(TARGET_DIR)/usr/share/games/doom/prboom.wad
