Add pyserial package to BR

Signed-off-by: Fabien Marteau <fabien.marteau@armadeus.com>

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-18 13:49:42.000000000 +0200
+++ buildroot/package/Config.in	2010-06-18 15:01:38.000000000 +0200
@@ -462,6 +462,7 @@
 source "package/lua/Config.in"
 source "package/microperl/Config.in"
 source "package/python/Config.in"
+source "package/pyserial/Config.in"
 source "package/ruby/Config.in"
 source "package/tcl/Config.in"
 source "package/php/Config.in"
Index: buildroot/package/pyserial/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/pyserial/Config.in	2010-06-18 14:16:37.000000000 +0200
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_PYSERIAL
+	bool "pyserial"
+	depends on BR2_PACKAGE_PYTHON
+	help
+	  Installs the python pyserial extension
Index: buildroot/package/pyserial/pyserial.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/pyserial/pyserial.mk	2010-06-18 15:07:52.000000000 +0200
@@ -0,0 +1,33 @@
+PYSERIAL_VERSION=2.2
+PYSERIAL_SOURCE=pyserial-$(PYSERIAL_VERSION).zip
+PYSERIAL_SITE=http://$(BR2_SOURCEFORGE_MIRROR).dl.sourceforge.net/sourceforge/pyserial
+PYSERIAL_DIR=$(BUILD_DIR)/pyserial-$(PYSERIAL_VERSION)
+PYSERIAL_TARGET_PKG=$(TARGET_DIR)/usr/lib/python2.4/site-packages/serial
+PYSERIAL_TARGET_BINARY=$(PYSERIAL_TARGET_PKG)/serialutil.py
+
+$(DL_DIR)/$(PYSERIAL_SOURCE):
+	$(call DOWNLOAD,$(PYSERIAL_SITE),$(PYSERIAL_SOURCE))
+
+$(PYSERIAL_DIR)/.unpacked: $(DL_DIR)/$(PYSERIAL_SOURCE)
+	unzip -o $(DL_DIR)/$(PYSERIAL_SOURCE) -d $(BUILD_DIR)
+	touch $@
+
+$(PYSERIAL_TARGET_BINARY): $(PYSERIAL_DIR)/.unpacked
+	mkdir -p $(PYSERIAL_TARGET_PKG)
+	for module in __init__.py serialposix.py serialutil.py; do \
+		cp $(PYSERIAL_DIR)/serial/$$module $(PYSERIAL_TARGET_PKG)/ ; \
+		done
+
+pyserial-source: $(DL_DIR)/$(PYSERIAL_SOURCE)
+
+pyserial-clean:
+	rm -rf $(PYSERIAL_TARGET_PKG)
+
+pyserial-dirclean:
+	rm -rf $(PYSERIAL_DIR)
+
+pyserial: python $(PYSERIAL_TARGET_BINARY)
+
+ifeq ($(strip $(BR2_PACKAGE_PYSERIAL)),y)
+TARGETS+=pyserial
+endif
