--- buildroot/package/php/php.mk.org	2009-04-28 15:45:51.000000000 +0200
+++ buildroot/package/php/php.mk	2009-04-28 15:57:47.000000000 +0200
@@ -5,7 +5,7 @@
 #############################################################
 PHP_VER:=5.2.6
 PHP_SOURCE:=php-$(PHP_VER).tar.bz2
-PHP_SITE:=http://us.php.net/get/${PHP_SOURCE}/from/us2.php.net/mirror
+PHP_SITE:=http://museum.php.net/php5/$(PHP_SOURCE)
 PHP_DIR:=$(BUILD_DIR)/php-$(PHP_VER)
 PHP_CAT=$(BZCAT)
 PHP_DEPS=
