--- buildrootref/target/ubifs/ubifsroot.mk	2008-11-03 00:15:09.000000000 -0800
+++ buildroot/target/ubifs/ubifsroot.mk	2009-06-02 14:39:18.000000000 -0700
@@ -5,8 +5,8 @@
 #############################################################
 #MKFS_UBIFS_VERSION=2582f128dad78591bc3adcc87c343c690bb82e61
 #MKFS_UBIFS_URL=http://git.infradead.org/users/dedekind/mkfs.ubifs.git?a=snapshot;h=$(MKFS_UBIFS_VERSION);sf=tgz
-MKFS_UBIFS_VERSION=v0.4
-MKFS_UBIFS_URL=http://git.infradead.org/users/dedekind/mkfs.ubifs.git?a=snapshot;h=refs/tags/mkfs.ubifs-$(MKFS_UBIFS_VERSION);sf=tgz
+MKFS_UBIFS_VERSION=HEAD
+MKFS_UBIFS_URL=http://git.infradead.org/users/ahunter/mkfs.ubifs.git?a=snapshot;h=$(MKFS_UBIFS_VERSION);sf=tgz
 MKFS_UBIFS_SOURCE:=mkfs.ubifs-$(MKFS_UBIFS_VERSION).tar.gz
 MKFS_UBIFS_DIR:= $(BUILD_DIR)/mkfs-ubifs-$(MKFS_UBIFS_VERSION)
 MKFS_UBIFS_CAT:=$(ZCAT)
@@ -107,6 +107,13 @@ endif
 	chmod a+x $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	$(STAGING_DIR)/usr/bin/fakeroot -- $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	-@rm -f $(PROJECT_BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
+	cp target/device/armadeus/$(BR2_BOARD_NAME)/ubinize.cfg .
+	cp $(BINARIES_DIR)/$(BR2_BOARD_NAME)-rootfs.arm.ubifs .
+	ubinize -o system_ubi.img -m $(BR2_TARGET_ROOTFS_UBIFS_MINIOSIZE) -p 128KiB -s 512 ubinize.cfg
+	cp -f system_ubi.img $(BINARIES_DIR)/$(BR2_BOARD_NAME)-rootfs.arm.ubifs
+	rm system_ubi.img
+	rm $(BR2_BOARD_NAME)-rootfs.arm.ubifs
+	rm ubinize.cfg
 
 ifneq ($(UBIFS_ROOTFS_COMPRESSOR),)
 $(UBIFS_BASE).$(UBIFS_ROOTFS_COMPRESSOR_EXT): $(UBIFS_ROOTFS_COMPRESSOR_PREREQ) $(UBIFS_BASE)
