To be removed when updating to latest Buildroot.

Index: buildroot/package/mtd/mtd-utils.git/mtd.mk
===================================================================
--- buildroot.orig/package/mtd/mtd-utils.git/mtd.mk	2009-12-17 15:54:31.000000000 +0100
+++ buildroot/package/mtd/mtd-utils.git/mtd.mk	2009-12-17 15:55:40.000000000 +0100
@@ -4,12 +4,14 @@
 #
 #############################################################
 MTD_VERSION:=fcb52ccc99679460640386c297023f852b108f68
+# gitweb >= 1.6.6-rc1 uses first 7 chars of git SHA1 as dir suffix
+MTD_SHORT_VERSION=$(shell echo $(MTD_VERSION)|cut -c -7)
 MTD_SOURCE:=mtd-utils-$(MTD_VERSION).tar.gz
 MTD_URL:=http://git.infradead.org/mtd-utils.git?a=snapshot;h=$(MTD_VERSION);sf=tgz
 MTD_HOST_DIR:= $(TOOL_BUILD_DIR)/mtd-utils-$(MTD_VERSION)
 MTD_DIR:=$(BUILD_DIR)/mtd-utils-$(MTD_VERSION)
 MTD_CAT:=$(ZCAT)
-MTD_NAME:=mtd-utils
+MTD_NAME:=mtd-utils-$(MTD_SHORT_VERSION)
 
 #############################################################
 #
