Already done in latest Buildroot version.

Index: buildroot/Makefile
===================================================================
--- buildroot.orig/Makefile	2010-02-02 16:23:20.000000000 +0100
+++ buildroot/Makefile	2010-02-02 16:26:34.000000000 +0100
@@ -21,7 +21,8 @@
 # Just run 'make menuconfig', configure stuff, then run 'make'.
 # You shouldn't need to mess with anything beyond this point...
 #--------------------------------------------------------------
-TOPDIR=./
+# absolute path
+TOPDIR:=$(shell pwd)
 CONFIG_CONFIG_IN=Config.in
 CONFIG_DEFCONFIG=.defconfig
 CONFIG=package/config
