Index: buildroot/package/audio/aumix/aumix.mk
===================================================================
--- buildroot.orig/package/audio/aumix/aumix.mk	2010-05-20 17:41:16.000000000 +0200
+++ buildroot/package/audio/aumix/aumix.mk	2010-05-20 17:42:03.000000000 +0200
@@ -6,7 +6,7 @@
 
 AUMIX_VERSION=2.8
 AUMIX_SOURCE=aumix-$(AUMIX_VERSION).tar.bz2
-AUMIX_SITE=http://jpj.net/~trevor/aumix
+AUMIX_SITE=http://jpj.net/~trevor/aumix/releases/
 AUMIX_DIR=$(BUILD_DIR)/aumix-$(AUMIX_VERSION)
 AUMIX_CAT:=$(BZCAT)
 
