Add package PPS Test scripts to Buildroot.

Signed-off-by: Jérémie Scheer <jeremie.scheer@armadeus.com>

Index: buildroot/package/armadeus/Config.in
===================================================================
--- buildroot.orig/package/armadeus/Config.in	2010-11-05 17:04:08.000000000 +0100
+++ buildroot/package/armadeus/Config.in	2010-11-05 17:05:06.000000000 +0100
@@ -8,6 +8,7 @@
 source package/armadeus/imxregs/Config.in
 source package/armadeus/imxssi/Config.in
 source package/armadeus/max5821ctrl/Config.in
+source package/armadeus/pps/Config.in
 source package/armadeus/testsuite/Config.in
 
 endmenu
Index: buildroot/package/armadeus/pps/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/armadeus/pps/Config.in	2010-11-05 17:05:39.000000000 +0100
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_ARMADEUS_PPS_TEST_SCRIPTS
+	bool "PPS Test scripts"
+	default n
+	help
+	  Install PPS Test scripts to test ethernet, wifi & USB Gadget on PPS board.
+	  Test scripts will be installed on the APF in /usr/local/pps/scripts.
Index: buildroot/package/armadeus/pps/pps.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/armadeus/pps/pps.mk	2010-11-08 14:09:58.000000000 +0100
@@ -0,0 +1,38 @@
+#############################################################
+#
+# PPS Test scripts
+#
+#############################################################
+
+ARMADEUS-PPS_TESTSCRIPTS_SOURCE:=$(TOPDIR)/../target/pps/scripts
+ARMADEUS-PPS_TESTSCRIPTS_TARGET_DIR:=$(TARGET_DIR)/usr/local/pps/scripts
+ARMADEUS-PPS_WWW_DIR:=$(TOPDIR)/../target/pps/www
+ARMADEUS-PPS_WWW_TARGET_DIR:=$(TARGET_DIR)/var
+ARMADEUS-PPS_CONFIGS_DIR:=$(TOPDIR)/../target/pps/configs
+ARMADEUS-PPS_CONFIGS_TARGET_DIR:=$(TARGET_DIR)/etc/wpa_supplicant
+
+$(ARMADEUS-PPS_TESTSCRIPTS_TARGET_DIR)/init_boa.sh:
+	$(ARMADEUS-PPS_TESTSCRIPTS_SOURCE)/install.sh $(ARMADEUS-PPS_TESTSCRIPTS_TARGET_DIR)
+
+$(ARMADEUS-PPS_WWW_TARGET_DIR)/www/index.html:
+	cp -Rf $(ARMADEUS-PPS_WWW_DIR) $(ARMADEUS-PPS_WWW_TARGET_DIR)
+
+$(ARMADEUS-PPS_CONFIGS_TARGET_DIR)/networkConfig:
+	mkdir -p $(ARMADEUS-PPS_CONFIGS_TARGET_DIR)
+	cp -f $(ARMADEUS-PPS_CONFIGS_DIR)/* $(ARMADEUS-PPS_CONFIGS_TARGET_DIR)
+
+pps-testscripts: $(ARMADEUS-PPS_TESTSCRIPTS_TARGET_DIR)/init_boa.sh $(ARMADEUS-PPS_WWW_TARGET_DIR)/www/index.html $(ARMADEUS-PPS_CONFIGS_TARGET_DIR)/networkConfig
+
+armadeus-pps-testscripts-clean:
+	rm -rf $(ARMADEUS-PPS_TESTSCRIPTS_TARGET_DIR)
+	rm -rf $(ARMADEUS-PPS_WWW_TARGET_DIR)/www
+	rm -rf $(ARMADEUS-PPS_CONFIGS_TARGET_DIR)
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_ARMADEUS_PPS_TEST_SCRIPTS)),y)
+TARGETS+=pps-testscripts
+endif
