--- gdb-6.4.50.20060212/bfd/sysdep.h	2005-05-05 20:51:14.000000000 +0200
+++ cygwin_sysdep.h	2006-10-04 18:39:00.328125000 +0200
@@ -77,7 +77,7 @@ extern char *strrchr ();
 #ifdef USE_BINARY_FOPEN
 #include "fopen-bin.h"
 #else
-#include "fopen-same.h"
+#include "../include/fopen-same.h"
 #endif
 
 #ifdef HAVE_FCNTL_H
@@ -108,7 +108,7 @@ extern char *strrchr ();
 #define SEEK_CUR 1
 #endif
 
-#include "filenames.h"
+#include "../include/filenames.h"
 
 #if !HAVE_DECL_FFS
 extern int ffs (int);
