diff -purN -x '*~' ref/u-boot-1.1.3/tools/Makefile u-boot-1.1.3/tools/Makefile
--- ref/u-boot-1.1.3/tools/Makefile	2006-01-27 01:25:42.000000000 +0100
+++ u-boot-1.1.3/tools/Makefile	2006-01-27 01:15:31.000000000 +0100
@@ -77,12 +77,12 @@ TOOLSUBDIRS =
 # multiple symbol definitions are treated as errors, hence the
 # -multiply_defined suppress option to turn off this error.
 #
-ifeq ($(HOSTOS)-$(HOSTARCH),darwin-ppc)
-HOST_CFLAGS = -traditional-cpp -Wall
-HOST_LDFLAGS =-multiply_defined suppress
-HOST_ENVIRO_CFLAGS = -traditional-cpp
+#ifeq ($(HOSTOS)-$(HOSTARCH),darwin-ppc)
+#HOST_CFLAGS = -traditional-cpp -Wall
+#HOST_LDFLAGS =-multiply_defined suppress
+#HOST_ENVIRO_CFLAGS = -traditional-cpp
 
-else
+#else
 ifeq ($(HOSTOS)-$(HOSTARCH),netbsd-ppc)
 HOST_CFLAGS = -Wall -pedantic
 HOST_LDFLAGS =
@@ -96,7 +96,7 @@ HOST_CFLAGS = -Wall -pedantic
 HOST_LDFLAGS =
 HOST_ENVIRO_CFLAGS =
 endif
-endif
+#endif
 
 #
 # Cygwin needs .exe files :-(
Binary files ref/u-boot-1.1.3/tools/mkbrecimage and u-boot-1.1.3/tools/mkbrecimage differ
diff -purN -x '*~' ref/u-boot-1.1.3/tools/mkimage.c u-boot-1.1.3/tools/mkimage.c
--- ref/u-boot-1.1.3/tools/mkimage.c	2005-08-14 01:53:35.000000000 +0200
+++ u-boot-1.1.3/tools/mkimage.c	2006-01-27 01:17:43.000000000 +0100
@@ -440,7 +440,7 @@ NXTARG:		;
 	}
 
 	/* We're a bit of paranoid */
-#if defined(_POSIX_SYNCHRONIZED_IO) && !defined(__sun__) && !defined(__FreeBSD__)
+#if defined(_POSIX_SYNCHRONIZED_IO) && !defined(__sun__) && !defined(__FreeBSD__) && !defined(__APPLE__)
 	(void) fdatasync (ifd);
 #else
 	(void) fsync (ifd);
@@ -490,7 +490,7 @@ NXTARG:		;
 	(void) munmap((void *)ptr, sbuf.st_size);
 
 	/* We're a bit of paranoid */
-#if defined(_POSIX_SYNCHRONIZED_IO) && !defined(__sun__) && !defined(__FreeBSD__)
+#if defined(_POSIX_SYNCHRONIZED_IO) && !defined(__sun__) && !defined(__FreeBSD__) && !defined(__APPLE__)
 	(void) fdatasync (ifd);
 #else
 	(void) fsync (ifd);
