--- linux-org/arch/arm/mach-imx/irq.c	2006-09-20 05:42:06.000000000 +0200
+++ linux-imx/arch/arm/mach-imx/irq.c	2006-12-09 17:37:53.386928000 +0100
@@ -67,6 +67,12 @@ imx_unmask_irq(unsigned int irq)
 {
 	__raw_writel(irq, IMX_AITC_INTENNUM);
 }
+
+static int
+imx_irq_type(unsigned int _irq, unsigned int type)
+{
+    return 0;
+}
 
 static int
 imx_gpio_irq_type(unsigned int _irq, unsigned int type)
@@ -209,6 +215,7 @@ static struct irq_chip imx_internal_chip
 	.ack = imx_mask_irq,
 	.mask = imx_mask_irq,
 	.unmask = imx_unmask_irq,
+	.set_type = imx_irq_type,
 };
 
