Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-01-25 10:04:10.000000000 +0100
+++ buildroot/package/Config.in	2010-01-25 10:04:25.000000000 +0100
@@ -316,6 +316,7 @@
 source "package/sdl_net/Config.in"
 source "package/sdl_ttf/Config.in"
 source "package/sdl_gfx/Config.in"
+source "package/slideshow/Config.in"
 source "package/tiff/Config.in"
 comment "Demos"
 source "package/sdlwater/Config.in"
