--- buildroot/package/fakeroot/fakeroot.mk.ref	2008-11-03 09:15:23.000000000 +0100
+++ buildroot/package/fakeroot/fakeroot.mk	2008-12-02 10:39:42.000000000 +0100
@@ -3,7 +3,7 @@
 # fakeroot
 #
 #############################################################
-FAKEROOT_VERSION:=1.9.5
+FAKEROOT_VERSION:=1.11
 FAKEROOT_SOURCE:=fakeroot_$(FAKEROOT_VERSION).tar.gz
 FAKEROOT_SITE:=http://snapshot.debian.net/archive/2008/04/27/debian/pool/main/f/fakeroot/
 FAKEROOT_CAT:=$(ZCAT)
