--- buildroot/target/linux/Makefile.in.advanced	2007-10-27 09:15:08.000000000 +0200
+++ buildroot/target/linux/Makefile.in.advanced.new	2008-10-22 16:06:21.000000000 +0200
@@ -43,7 +43,8 @@ endif
 endif

 # Ask the kernel what version number it will use when installing modules
-LINUX26_VERSION_PROBED = `$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) -s kernelrelease`
+LINUX26_VERSION_PROBED = $(LINUX26_VERSION)
+#`$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) -s kernelrelease`

 ifeq ($(BOARD_PATH),)
 BOARD_PATH:=$(call qstrip,$(BR2_BOARD_PATH))
@@ -159,8 +159,9 @@ endif
 ROOTFS_STRIPPED_SUFFIX=$(call qstrip,$(ROOTFS_SUFFIX))
 # -----------------------------------------------------------------------------
 # File name for the Linux kernel binary
+ifeq ($(strip $(LINUX26_KERNEL_NAME)),)
 LINUX26_KERNEL_NAME:=$(LINUX26_FORMAT)$(ROOTFS_STRIPPED_SUFFIX)
-
+endif

 ifeq ($(BOARD_NAME),)
 BOARD_NAME=$(ARCH)
@@ -280,8 +282,8 @@ endif
 	$(SED) '/CONFIG_AEABI=y/d' $(LINUX26_DIR)/.config
 ifeq ($(BR2_ARM_EABI),y)
 	echo "CONFIG_AEABI=y" >> $(LINUX26_DIR)/.config
-	$(SED) '/CONFIG_OABI_COMPAT/d' $(LINUX26_DIR)/.config
-	echo "# CONFIG_OABI_COMPAT is not set" >> $(LINUX26_DIR)/.config
+#	$(SED) '/CONFIG_OABI_COMPAT/d' $(LINUX26_DIR)/.config
+#	echo "# CONFIG_OABI_COMPAT is not set" >> $(LINUX26_DIR)/.config
 else
 	echo "# CONFIG_AEABI is not set" >> $(LINUX26_DIR)/.config
 endif
@@ -310,6 +312,9 @@ ifeq ($(BR2_PACKAGE_BUSYBOX_INITRAMFS),y
 	echo "CONFIG_INITRAMFS_ROOT_UID=0" >> $(LINUX26_DIR)/.config
 	echo "CONFIG_INITRAMFS_ROOT_GID=0" >> $(LINUX26_DIR)/.config
 endif
+ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
+	(cd $(LINUX_DIR)/drivers; ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ armadeus)
+endif
 	$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) oldconfig
 	touch $@
 
