--- buildroot/toolchain/uClibc/uclibc.mk.ref	2008-10-24 11:13:56.000000000 +0200
+++ buildroot/toolchain/uClibc/uclibc.mk	2008-11-05 21:30:07.000000000 +0100
@@ -141,23 +141,6 @@ $(UCLIBC_DIR)/.oldconfig: $(UCLIBC_DIR)/
 		-e 's,^SHARED_LIB_LOADER_PREFIX=.*,SHARED_LIB_LOADER_PREFIX=\"/lib\",g' \
 		$(UCLIBC_DIR)/.oldconfig
 ifeq ($(UCLIBC_TARGET_ARCH),arm)
-	(/bin/echo "# CONFIG_GENERIC_ARM is not set"; \
-	 /bin/echo "# CONFIG_ARM610 is not set"; \
-	 /bin/echo "# CONFIG_ARM710 is not set"; \
-	 /bin/echo "# CONFIG_ARM7TDMI is not set"; \
-	 /bin/echo "# CONFIG_ARM720T is not set"; \
-	 /bin/echo "# CONFIG_ARM920T is not set"; \
-	 /bin/echo "# CONFIG_ARM922T is not set"; \
-	 /bin/echo "# CONFIG_ARM926T is not set"; \
-	 /bin/echo "# CONFIG_ARM10T is not set"; \
-	 /bin/echo "# CONFIG_ARM1136JF_S is not set"; \
-	 /bin/echo "# CONFIG_ARM1176JZ_S is not set"; \
-	 /bin/echo "# CONFIG_ARM1176JZF_S is not set"; \
-	 /bin/echo "# CONFIG_ARM_SA110 is not set"; \
-	 /bin/echo "# CONFIG_ARM_SA1100 is not set"; \
-	 /bin/echo "# CONFIG_ARM_XSCALE is not set"; \
-	 /bin/echo "# CONFIG_ARM_IWMMXT is not set"; \
-	) >> $(UCLIBC_DIR)/.oldconfig
 	$(SED) 's/^\(CONFIG_[^_]*[_]*ARM[^=]*\)=.*/# \1 is not set/g' \
 		 $(UCLIBC_DIR)/.oldconfig
 	$(SED) 's/^.*$(UCLIBC_ARM_TYPE).*/$(UCLIBC_ARM_TYPE)=y/g' $(UCLIBC_DIR)/.oldconfig
