--- buildroot/target/linux/Makefile.in.advanced.org	2008-12-12 21:55:37.000000000 +0100
+++ buildroot/target/linux/Makefile.in.advanced	2008-12-12 22:25:20.000000000 +0100
@@ -378,6 +378,8 @@
 	echo "CONFIG_INITRAMFS_ROOT_GID=0" >> $(LINUX26_DIR)/.config
 endif
 ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
+	$(SED) "s,^ARCH.*,ARCH=$(KERNEL_ARCH),g;" $(LINUX_DIR)/Makefile
+	$(SED) "s,^CROSS_COMPILE.*,CROSS_COMPILE=$(KERNEL_CROSS),g;" $(LINUX_DIR)/Makefile
 	(cd $(LINUX_DIR)/drivers; ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ armadeus)
 endif
 	$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) oldconfig
