--- buildroot/target/linux/Makefile.in.advanced.org	2009-02-18 09:21:59.000000000 +0100
+++ buildroot/target/linux/Makefile.in.advanced	2009-02-18 09:35:47.000000000 +0100
@@ -293,7 +293,7 @@
 ifneq ($(LINUX26_PATCH_DIR),)
 	if [ -d $(LINUX26_PATCH_DIR) ] ; then \
 	toolchain/patch-kernel.sh $(LINUX26_DIR) $(LINUX26_PATCH_DIR) \
-		linux-$(LINUX26_VERSION)\*.patch{,.gz,.bz2} ; \
+		\*.patch{,.gz,.bz2} ; \
 	fi
 endif
 	touch $@
