--- buildrootref/target/ubifs/Config.in	2008-11-03 09:15:09.000000000 +0100
+++ buildroot/target/ubifs/Config.in	2009-10-03 17:07:32.000000000 +0200
@@ -1,10 +1,15 @@
 config BR2_TARGET_ROOTFS_UBIFS
 	bool "ubifs root filesystem"
-	depends on BROKEN # upstream git gone, should use mtd-utils
 	select BR2_HOST_FAKEROOT
+	select BR2_PACKAGE_MTD
 	help
 	  Build a ubifs root filesystem
 
+config BR2_TARGET_ROOTFS_UBIFS_PEBSIZE
+	hex "UBI physical erase block size"
+	depends on BR2_TARGET_ROOTFS_UBIFS
+	default 0x20000
+
 config BR2_TARGET_ROOTFS_UBIFS_LEBSIZE
 	hex "UBI logical erase block size"
 	depends on BR2_TARGET_ROOTFS_UBIFS
@@ -17,6 +23,13 @@ config BR2_TARGET_ROOTFS_UBIFS_MINIOSIZE
 	help
 	  Some comment required here
 
+config BR2_TARGET_ROOTFS_UBIFS_SUBSIZE
+	int "UBI sub-page size"
+	depends on BR2_TARGET_ROOTFS_UBIFS
+	default 512
+	help
+	  Some comment required here
+
 config BR2_TARGET_ROOTFS_UBIFS_MAXLEBCNT
 	int "Maximum LEB count"
 	depends on BR2_TARGET_ROOTFS_UBIFS
--- buildrootref/target/ubifs/ubifsroot.mk	2009-10-03 17:55:26.000000000 +0200
+++ buildroot/target/ubifs/ubifsroot.mk	2009-10-03 17:18:09.000000000 +0200
@@ -75,7 +75,14 @@ else
 UBIFS_TARGET := $(UBIFS_BASE)
 endif
 
-$(UBIFS_BASE): host-fakeroot makedevs mkfs.ubifs
+UBIFS_UBINIZE_OPTS := -m $(BR2_TARGET_ROOTFS_UBIFS_MINIOSIZE) -p $(BR2_TARGET_ROOTFS_UBIFS_PEBSIZE)
+
+ifneq ($(BR2_TARGET_ROOTFS_UBIFS_SUBSIZE),0)
+UBIFS_UBINIZE_OPTS += -s $(BR2_TARGET_ROOTFS_UBIFS_SUBSIZE)
+endif
+
+
+$(UBIFS_BASE): host-fakeroot makedevs
 	# Use fakeroot to pretend all target binaries are owned by root
 	rm -f $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	touch $(BUILD_DIR)/.fakeroot.00000
@@ -80,11 +87,18 @@ ifneq ($(TARGET_DEVICE_TABLE),)
 		>> $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 endif
 	# Use fakeroot so mkfs.ubifs believes the previous fakery
-	echo "$(MKFS_UBIFS_DIR)/mkfs.ubifs -d $(TARGET_DIR) " \
+	echo "$(MKFS_UBIFS) -d $(TARGET_DIR) " \
 		"$(UBIFS_OPTS) -o $(UBIFS_BASE)" >> $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	chmod a+x $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	$(HOST_DIR)/usr/bin/fakeroot -- $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
 	-@rm -f $(BUILD_DIR)/_fakeroot.$(notdir $(UBIFS_TARGET))
+	cp target/ubifs/ubinize.cfg .
+	echo "image=$(UBIFS_BASE)" \
+		>> ./ubinize.cfg
+	$(UBINIZE) -o system_ubi.img $(UBIFS_UBINIZE_OPTS) ubinize.cfg
+	cp -f system_ubi.img $(UBIFS_BASE)
+	rm system_ubi.img
+	rm ubinize.cfg
 
 ifneq ($(UBIFS_ROOTFS_COMPRESSOR),)
 $(UBIFS_BASE).$(UBIFS_ROOTFS_COMPRESSOR_EXT): $(UBIFS_ROOTFS_COMPRESSOR_PREREQ) $(UBIFS_BASE)
