--- MPlayer-1.0rc1/libavcodec/armv4l/dsputil_arm_s.S.ori	2006-10-23 00:32:33.000000000 +0200
+++ MPlayer-1.0rc1/libavcodec/armv4l/dsputil_arm_s.S	2009-01-08 00:26:45.000000000 +0100
@@ -19,6 +19,11 @@
 @ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 @
 
+#if !defined(HAVE_ARMV5TE) && !defined(HAVE_IWMMXT)
+.macro pld arg
+        @
+.endm
+#endif
 .macro  ADJ_ALIGN_QUADWORD_D shift, Rd0, Rd1, Rd2, Rd3, Rn0, Rn1, Rn2, Rn3, Rn4
         mov \Rd0, \Rn0, lsr #(\shift * 8)
         mov \Rd1, \Rn1, lsr #(\shift * 8)
