diff -purN -x '*~' ref/u-boot-1.1.6/common/spartan2.c u-boot-1.1.6/common/spartan2.c
--- ref/u-boot-1.1.6/common/spartan2.c	2006-11-02 15:15:01.000000000 +0100
+++ u-boot-1.1.6/common/spartan2.c	2006-11-25 22:48:54.000000000 +0100
@@ -441,7 +441,7 @@ static int Spartan2_ss_load (Xilinx_desc
 	int ret_val = FPGA_FAIL;	/* assume the worst */
 	Xilinx_Spartan2_Slave_Serial_fns *fn = desc->iface_fns;
 	int i;
-	char  val;
+	signed char  val;
 
 	PRINTF ("%s: start with interface functions @ 0x%p\n",
 			__FUNCTION__, fn);
diff -purN -x '*~' ref/u-boot-1.1.6/common/spartan3.c u-boot-1.1.6/common/spartan3.c
--- ref/u-boot-1.1.6/common/spartan3.c	2006-11-02 15:15:01.000000000 +0100
+++ u-boot-1.1.6/common/spartan3.c	2006-11-25 22:48:58.000000000 +0100
@@ -446,7 +446,7 @@ static int Spartan3_ss_load (Xilinx_desc
 	int ret_val = FPGA_FAIL;	/* assume the worst */
 	Xilinx_Spartan3_Slave_Serial_fns *fn = desc->iface_fns;
 	int i;
-	char  val;
+	signed char  val;
 
 	PRINTF ("%s: start with interface functions @ 0x%p\n",
 			__FUNCTION__, fn);
