--- u-boot-2010.03-rc1.ori/MAKEALL	2010-03-12 23:06:04.000000000 +0100
+++ u-boot-2010.03-rc1/MAKEALL	2010-03-23 01:33:00.000000000 +0100
@@ -543,6 +543,7 @@ LIST_ARM7="		\
 
 LIST_ARM9="			\
 	a320evb			\
+	apf9328			\
 	ap920t			\
 	ap922_XA10		\
 	ap926ejs		\
--- u-boot-2010.03-rc1.ori/Makefile	2010-03-12 23:06:04.000000000 +0100
+++ u-boot-2010.03-rc1/Makefile	2010-03-23 01:26:08.000000000 +0100
@@ -2689,6 +2689,9 @@ a320evb_config	:	unconfig
 ## Atmel AT91RM9200 Systems
 #########################################################################
 
+apf9328_config		:	unconfig
+	@$(MKCONFIG) $(@:_config=) arm arm920t apf9328 armadeus imx
+
 at91rm9200dk_config	:	unconfig
 	@$(MKCONFIG) $(@:_config=) arm arm920t at91rm9200dk atmel at91rm9200
 
