--- u-boot-2010.03/fs/jffs2/jffs2_1pass.c.ori	2010-04-06 22:42:11.000000000 +0200
+++ u-boot-2010.03/fs/jffs2/jffs2_1pass.c	2010-04-06 22:42:20.000000000 +0200
@@ -196,8 +196,8 @@ static int read_nand_cached(u32 off, u32
 			if (nand_read(&nand_info[id->num], nand_cache_off,
 						&retlen, nand_cache) != 0 ||
 					retlen != NAND_CACHE_SIZE) {
-				printf("read_nand_cached: error reading nand off %#x size %d bytes\n",
-						nand_cache_off, NAND_CACHE_SIZE);
+/*				printf("read_nand_cached: error reading nand off %#x size %d bytes\n",
+						nand_cache_off, NAND_CACHE_SIZE);*/
 				return -1;
 			}
 		}
