diff -purN linux-2.6.12-ref/arch/arm/mach-imx/scb9328.c linux-2.6.12/arch/arm/mach-imx/scb9328.c
--- linux-2.6.12-ref/arch/arm/mach-imx/scb9328.c	2005-12-21 17:56:21.000000000 +0100
+++ linux-2.6.12/arch/arm/mach-imx/scb9328.c	2005-08-10 22:57:32.000000000 +0200
@@ -258,7 +258,7 @@ static struct platform_device *devices[]
 
 static int __init scb9328_init(void)
 {
-	evk9328_init_leds();
+/// ej	//evk9328_init_leds();
 	UFCR(IMX_UART1_BASE) = 0xa81;
 	UFCR(IMX_UART2_BASE) = 0xa81;
 	return platform_add_devices(devices, ARRAY_SIZE(devices));
@@ -290,6 +290,7 @@ static struct map_desc scb9328_io_desc[]
 	{IMX_CS1_VIRT, IMX_CS1_PHYS, IMX_CS1_SIZE, MT_DEVICE},
 	{IMX_CS2_VIRT, IMX_CS2_PHYS, IMX_CS2_SIZE, MT_DEVICE},
 	{IMX_CS3_VIRT, IMX_CS3_PHYS, IMX_CS3_SIZE, MT_DEVICE},
+	{IMX_CS4_VIRT, IMX_CS4_PHYS, IMX_CS4_SIZE, MT_DEVICE},
 	{IMX_CS5_VIRT, IMX_CS5_PHYS, IMX_CS5_SIZE, MT_DEVICE},
 };
 
@@ -298,7 +299,7 @@ static void __init scb9328_init_irq(void
 	imx_init_irq();
 	set_irq_type(SCB9328_UART1_IRQ, IRQT_LOW);
 #ifdef CONFIG_DM9000
-	set_irq_type(SCB9328_ETH_IRQ, IRQT_LOW);
+	set_irq_type(SCB9328_ETH_IRQ, /*IRQT_LOW*/ IRQT_HIGH);
 #endif
 }

diff -purN linux-2.6.12-ref/include/asm-arm/arch-imx/scb9328.h linux-2.6.12/include/asm-arm/arch-imx/scb9328.h
--- linux-2.6.12-ref/include/asm-arm/arch-imx/scb9328.h	2005-12-21 17:56:21.000000000 +0100
+++ linux-2.6.12/include/asm-arm/arch-imx/scb9328.h	2005-08-09 08:39:04.000000000 +0200
@@ -27,9 +27,13 @@
 #define SCB9328_UART2_PHYS  ( IMX_CS3_PHYS + 0x10000 )
 #define SCB9328_UART2_IRQ   IRQ_GPIOA(1)
 
-#define SCB9328_ETH_VIRT ( IMX_CS5_VIRT )
-#define SCB9328_ETH_PHYS ( IMX_CS5_PHYS )
-#define SCB9328_ETH_IRQ  ( IRQ_GPIOC(3) )
+/* #define SCB9328_ETH_VIRT ( IMX_CS5_VIRT )*/
+/* #define SCB9328_ETH_PHYS ( IMX_CS5_PHYS )*/
+/* #define SCB9328_ETH_IRQ  ( IRQ_GPIOC(3) )*/
+
+#define SCB9328_ETH_VIRT ( IMX_CS4_VIRT + 0x0C3FFFC )
+#define SCB9328_ETH_PHYS ( IMX_CS4_PHYS + 0x0C3FFFC )
+#define SCB9328_ETH_IRQ  ( IRQ_GPIOB(14) )
 
 #define SCB9328_FLASH_PHYS	IMX_CS0_PHYS
 #define SCB9328_FLASH_SIZE	0x01000000
