Index: buildroot/target/device/Config.in
===================================================================
--- buildroot.orig/target/device/Config.in	2010-06-17 16:45:14.000000000 +0200
+++ buildroot/target/device/Config.in	2010-06-17 16:45:25.000000000 +0200
@@ -2,6 +2,7 @@
 
 comment "Preset Devices"
 
+source "target/device/armadeus/Config.in"
 source "target/device/ARMLTD/Config.in"
 source "target/device/Atmel/Config.in"
 source "target/device/KwikByte/Config.in"
