Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-17 16:45:46.000000000 +0200
+++ buildroot/package/Config.in	2010-06-17 16:45:48.000000000 +0200
@@ -86,6 +86,7 @@
 menu "Graphic libraries and applications (graphic/text)"
 
 comment "graphic libraries"
+source "package/allegro/Config.in"
 source "package/directfb/Config.in"
 source "package/directfb-examples/Config.in"
 source "package/divine/Config.in"
