Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-01-25 10:04:00.000000000 +0100
+++ buildroot/package/Config.in	2010-01-25 10:04:02.000000000 +0100
@@ -86,6 +86,7 @@
 menu "Graphic libraries and applications (graphic/text)"
 
 comment "graphic libraries"
+source "package/allegro/Config.in"
 source "package/directfb/Config.in"
 source "package/directfb-examples/Config.in"
 source "package/divine/Config.in"
