Index: buildroot/package/fakeroot/fakeroot.mk
===================================================================
--- buildroot.orig/package/fakeroot/fakeroot.mk	2010-06-17 16:45:13.000000000 +0200
+++ buildroot/package/fakeroot/fakeroot.mk	2010-06-17 16:46:25.000000000 +0200
@@ -3,7 +3,7 @@
 # fakeroot
 #
 #############################################################
-FAKEROOT_VERSION:=1.9.5
+FAKEROOT_VERSION:=1.11
 FAKEROOT_SOURCE:=fakeroot_$(FAKEROOT_VERSION).tar.gz
 FAKEROOT_SITE:=http://snapshot.debian.net/archive/2008/04/27/debian/pool/main/f/fakeroot/
 FAKEROOT_LIBTOOL_PATCH=NO
