Index: buildroot/target/linux/Makefile.in.advanced
===================================================================
--- buildroot.orig/target/linux/Makefile.in.advanced	2010-06-17 16:45:13.000000000 +0200
+++ buildroot/target/linux/Makefile.in.advanced	2010-06-17 16:46:27.000000000 +0200
@@ -41,7 +41,8 @@
 endif
 
 # Ask the kernel what version number it will use when installing modules
-LINUX26_VERSION_PROBED = `$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) --no-print-directory -s kernelrelease`
+LINUX26_VERSION_PROBED = $(LINUX26_VERSION)
+#`$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) --no-print-directory -s kernelrelease`
 
 ifeq ($(BOARD_PATH),)
 BOARD_PATH:=$(call qstrip,$(BR2_BOARD_PATH))
@@ -162,8 +163,9 @@
 
 # -----------------------------------------------------------------------------
 # File name for the Linux kernel binary
+ifeq ($(strip $(LINUX26_KERNEL_NAME)),)
 LINUX26_KERNEL_NAME:=$(LINUX26_FORMAT)
-
+endif
 
 ifeq ($(BOARD_NAME),)
 BOARD_NAME=$(ARCH)
@@ -334,8 +336,8 @@
 	$(SED) '/CONFIG_AEABI=y/d' $(LINUX26_DIR)/.config
 ifeq ($(BR2_ARM_EABI),y)
 	echo "CONFIG_AEABI=y" >> $(LINUX26_DIR)/.config
-	$(SED) '/CONFIG_OABI_COMPAT/d' $(LINUX26_DIR)/.config
-	echo "# CONFIG_OABI_COMPAT is not set" >> $(LINUX26_DIR)/.config
+#	$(SED) '/CONFIG_OABI_COMPAT/d' $(LINUX26_DIR)/.config
+#	echo "# CONFIG_OABI_COMPAT is not set" >> $(LINUX26_DIR)/.config
 else
 	echo "# CONFIG_AEABI is not set" >> $(LINUX26_DIR)/.config
 endif
@@ -364,6 +366,9 @@
 	echo "CONFIG_INITRAMFS_ROOT_UID=0" >> $(LINUX26_DIR)/.config
 	echo "CONFIG_INITRAMFS_ROOT_GID=0" >> $(LINUX26_DIR)/.config
 endif
+ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
+	(cd $(LINUX_DIR)/drivers; ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ armadeus)
+endif
 	$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) oldconfig
 	touch $@
 
