Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-17 16:46:21.000000000 +0200
+++ buildroot/package/Config.in	2010-06-17 16:46:29.000000000 +0200
@@ -91,6 +91,7 @@
 source "package/directfb-examples/Config.in"
 source "package/divine/Config.in"
 source "package/fbdump/Config.in"
+source "package/fbgrab/Config.in"
 if BR2_PACKAGE_BUSYBOX_SHOW_OTHERS
 source "package/fbset/Config.in"
 endif
