Specific for Armadeus (needed by quiltify script).

Index: buildroot/target/u-boot/Makefile.in
===================================================================
--- buildroot.orig/target/u-boot/Makefile.in	2010-06-17 16:46:58.000000000 +0200
+++ buildroot/target/u-boot/Makefile.in	2010-06-17 16:47:00.000000000 +0200
@@ -74,6 +74,8 @@
 	mkdir -p $(U_BOOT_DIR)
 	touch $@
 
+u-boot-src-unpacked: $(U_BOOT_DIR)/.unpacked
+
 $(U_BOOT_DIR)/.patched: $(U_BOOT_DIR)/.unpacked
 	toolchain/patch-kernel.sh $(U_BOOT_DIR) target/u-boot \
 		u-boot-$(U_BOOT_VERSION)-\*.patch \
