Index: buildroot/target/linux/Makefile.in.advanced
===================================================================
--- buildroot.orig/target/linux/Makefile.in.advanced	2010-06-17 16:46:38.000000000 +0200
+++ buildroot/target/linux/Makefile.in.advanced	2010-06-17 16:47:06.000000000 +0200
@@ -468,7 +468,9 @@
 
 linux26-source: $(DL_DIR)/$(LINUX26_SOURCE) $(LINUX26_PATCH_SOURCE_FILE)
 
-linux26-unpacked: $(LINUX26_DIR)/.patched
+linux26-unpacked: $(LINUX26_DIR)/.unpacked
+
+linux26-patched: $(LINUX26_DIR)/.patched
 
 linux26-config: host-sed $(LINUX26_DIR)/.configured
 
