Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-17 16:46:31.000000000 +0200
+++ buildroot/package/Config.in	2010-06-17 16:47:12.000000000 +0200
@@ -216,6 +216,7 @@
 source "package/microperl/Config.in"
 source "package/php/Config.in"
 source "package/python/Config.in"
+source "package/pyserial/Config.in"
 source "package/ruby/Config.in"
 source "package/tcl/Config.in"
 endmenu
