Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-17 16:47:12.000000000 +0200
+++ buildroot/package/Config.in	2010-06-17 16:47:13.000000000 +0200
@@ -217,6 +217,7 @@
 source "package/php/Config.in"
 source "package/python/Config.in"
 source "package/pyserial/Config.in"
+source "package/pymysql/Config.in"
 source "package/ruby/Config.in"
 source "package/tcl/Config.in"
 endmenu
