Index: buildroot/linux/linux.mk
===================================================================
--- buildroot.orig/linux/linux.mk	2010-10-19 09:43:24.000000000 +0200
+++ buildroot/linux/linux.mk	2010-10-19 09:48:32.000000000 +0200
@@ -111,6 +111,8 @@
 	$(call KCONFIG_ENABLE_OPT,CONFIG_INITRAMFS_COMPRESSION_GZIP,$(@D)/.config)
 endif
 ifeq ($(strip $(BR2_TARGET_ARMADEUS)),y)
+	$(SED) "s,^ARCH.*,ARCH=$(KERNEL_ARCH),g;" $(LINUX26_DIR)/Makefile
+	$(SED) "s,^CROSS_COMPILE.*,CROSS_COMPILE=$(KERNEL_CROSS),g;" $(LINUX26_DIR)/Makefile
 	(cd $(LINUX26_DIR)/drivers; ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ armadeus)
 endif
 	$(TARGET_MAKE_ENV) $(MAKE) $(LINUX26_MAKE_FLAGS) -C $(@D) oldconfig
