Index: buildroot/linux/linux.mk
===================================================================
--- buildroot.orig/linux/linux.mk	2010-10-19 10:13:07.000000000 +0200
+++ buildroot/linux/linux.mk	2010-10-19 10:29:21.000000000 +0200
@@ -74,7 +74,7 @@
 ifneq ($(filter ftp://% http://%,$(LINUX26_PATCH)),)
 	toolchain/patch-kernel.sh $(@D) $(DL_DIR) $(notdir $(LINUX26_PATCH))
 else ifeq ($(shell test -d $(LINUX26_PATCH) && echo "dir"),dir)
-	toolchain/patch-kernel.sh $(@D) $(LINUX26_PATCH) linux-\*.patch
+	toolchain/patch-kernel.sh $(@D) $(LINUX26_PATCH) \*.patch
 else
 	toolchain/patch-kernel.sh $(@D) $(dir $(LINUX26_PATCH)) $(notdir $(LINUX26_PATCH))
 endif
