Index: buildroot/target/linux/Makefile.in.advanced
===================================================================
--- buildroot.orig/target/linux/Makefile.in.advanced	2010-06-17 16:47:15.000000000 +0200
+++ buildroot/target/linux/Makefile.in.advanced	2010-06-17 16:47:19.000000000 +0200
@@ -278,7 +278,7 @@
 ifneq ($(LINUX26_PATCH_DIR),)
 	if [ -d $(LINUX26_PATCH_DIR) ] ; then \
 	toolchain/patch-kernel.sh $(LINUX26_DIR) $(LINUX26_PATCH_DIR) \
-		linux-$(LINUX26_VERSION)\*.patch{,.gz,.bz2} ; \
+		\*.patch{,.gz,.bz2} ; \
 	fi
 endif
 	touch $@
