Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-17 16:47:13.000000000 +0200
+++ buildroot/package/Config.in	2010-06-17 16:47:20.000000000 +0200
@@ -206,6 +206,7 @@
 source "package/usbmount/Config.in"
 source "package/usbutils/Config.in"
 source "package/wipe/Config.in"
+source "package/xenomai/Config.in"
 source "package/xfsprogs/Config.in"
 endmenu
 
Index: buildroot/target/linux/Makefile.in.advanced
===================================================================
--- buildroot.orig/target/linux/Makefile.in.advanced	2010-06-17 16:47:19.000000000 +0200
+++ buildroot/target/linux/Makefile.in.advanced	2010-06-17 16:47:20.000000000 +0200
@@ -373,6 +373,9 @@
 	$(SED) "s,^CROSS_COMPILE.*,CROSS_COMPILE=$(KERNEL_CROSS),g;" $(LINUX_DIR)/Makefile
 	(cd $(LINUX_DIR)/drivers; ln -sf $(ARMADEUS_TOPDIR)/target/linux/modules/ armadeus)
 endif
+ifeq ($(BR2_PACKAGE_XENOMAI),y)
+	make -C ../buildroot xenomai-patch-kernel
+endif
 	$(MAKE) $(LINUX26_MAKE_FLAGS) -C $(LINUX26_DIR) oldconfig
 	touch $@
 
