From dc36961d7212902e74309320912f55950b9fc534 Mon Sep 17 00:00:00 2001
From: Julien Boibessot <julien.boibessot@armadeus.com>
Date: Thu, 20 May 2010 17:58:05 +0200
Subject: [PATCH] [PATCH] aumix SITE has changed


Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>
---
 package/multimedia/aumix/aumix.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/package/multimedia/aumix/aumix.mk b/package/multimedia/aumix/aumix.mk
index e28a720..65d9d39 100644
--- a/package/multimedia/aumix/aumix.mk
+++ b/package/multimedia/aumix/aumix.mk
@@ -6,7 +6,7 @@
 
 AUMIX_VERSION=2.8
 AUMIX_SOURCE=aumix-$(AUMIX_VERSION).tar.bz2
-AUMIX_SITE=http://jpj.net/~trevor/aumix
+AUMIX_SITE=http://jpj.net/~trevor/aumix/releases
 AUMIX_AUTORECONF=YES
 
 AUMIX_CONF_OPT = \
-- 
1.6.0.4

