From 133a3f90ba01f76568a8b8dec39fd5cf158dd494 Mon Sep 17 00:00:00 2001
From: Julien Boibessot <julien.boibessot@armadeus.com>
Date: Fri, 04 Jun 2010 15:21:04 +0000
Subject: Set correct GCC tune option for ARM926T(==ARM926EJ-S) core

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>
Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
---
diff --git a/target/Config.in.arch b/target/Config.in.arch
index 1de679f..62721c9 100644
--- a/target/Config.in.arch
+++ b/target/Config.in.arch
@@ -594,7 +594,7 @@ config BR2_GCC_TARGET_TUNE
 	default arm920		if BR2_arm920
 	default arm920t		if BR2_arm920t
 	default arm922t		if BR2_arm922t
-	default arm9tdmi	if BR2_arm926t
+	default arm926ej-s	if BR2_arm926t
 	default arm1136j-s	if BR2_arm1136j_s
 	default arm1136jf-s	if BR2_arm1136jf_s
 	default arm1176jz-s	if BR2_arm1176jz_s
--
cgit v0.8.2.1
