--- alsa-driver-1.0.14.ref/acinclude.m4	2007-06-04 09:28:55.000000000 +0200
+++ alsa-driver-1.0.14/acinclude.m4	2007-09-09 22:53:57.000000000 +0200
@@ -186,6 +186,8 @@ AC_DEFUN([ALSA_TOPLEVEL_INIT], [
 	CONFIG_SND_AOA_SOUNDBUS_I2S=""
 	CONFIG_ARM=""
 	CONFIG_SND_SA11XX_UDA1341=""
+	CONFIG_SND_IMX_SSI=""
+	CONFIG_SND_IMX_TSC2102=""
 	CONFIG_ARCH_SA1100=""
 	CONFIG_L3=""
 	CONFIG_SND_ARMAACI=""
@@ -299,7 +301,7 @@ AC_ARG_WITH(cards,
   [                          usb-usx2y, usb-caiaq, vxpocket, pdaudiocf, ]
   [                          sun-amd7930, sun-cs4231, sun-dbri, harmony, soc, ]
   [                          at91-soc, at91-soc-eti-b1-wm8731, pxa2xx-soc, ]
-  [                          pxa2xx-soc-corgi, pxa2xx-soc-spitz, ]
+  [                          pxa2xx-soc-corgi, pxa2xx-soc-spitz, imx-tsc2102, ]
   [                          pxa2xx-soc-poodle, pxa2xx-soc-tosa, s3c24xx-soc, ]
   [                          soc-pcm-sh7760, sh7760-ac97 ],
   cards="$withval", cards="all")
@@ -1962,6 +1964,19 @@ for card in $SELECTED_CARDS; do
       probed=0
     fi
   fi
+  if test "$card" = "all" -o "$card" = "imx-tsc2102"; then
+    if ( test "$CONFIG_SOUND" = "y" -o "$CONFIG_SOUND" = "m" ) &&
+      ( test "$CONFIG_ARM" = "y" -o "$CONFIG_ARM" = "m" ) &&
+      ( test "$CONFIG_ARCH_IMX" = "y" -o "$CONFIG_ARCH_IMX" = "m" ); then
+      CONFIG_SND_TIMER="m"
+      CONFIG_SND_PCM="m"
+      CONFIG_SND_IMX_TSC2102="m"
+      CONFIG_SND_IMX_SSI="m"
+      probed=1
+    elif test -z "$probed"; then
+      probed=0
+    fi
+  fi
   if test "$card" = "all" -o "$card" = "armaaci"; then
     if ( test "$CONFIG_SOUND" = "y" -o "$CONFIG_SOUND" = "m" ) &&
       ( test "$CONFIG_ARM" = "y" -o "$CONFIG_ARM" = "m" ) &&
@@ -2928,6 +2943,12 @@ fi
 if test -n "$CONFIG_SND_SA11XX_UDA1341"; then
   AC_DEFINE(CONFIG_SND_SA11XX_UDA1341_MODULE)
 fi
+if test -n "$CONFIG_SND_IMX_SSI"; then
+  AC_DEFINE(CONFIG_SND_IMX_SSI_MODULE)
+fi
+if test -n "$CONFIG_SND_IMX_TSC2102"; then
+  AC_DEFINE(CONFIG_SND_IMX_TSC2102_MODULE)
+fi
 if test -n "$CONFIG_SND_ARMAACI"; then
   AC_DEFINE(CONFIG_SND_ARMAACI_MODULE)
 fi
@@ -3306,6 +3327,8 @@ AC_SUBST(CONFIG_SND_AOA_SOUNDBUS)
 AC_SUBST(CONFIG_SND_AOA_SOUNDBUS_I2S)
 AC_SUBST(CONFIG_ARM)
 AC_SUBST(CONFIG_SND_SA11XX_UDA1341)
+AC_SUBST(CONFIG_SND_IMX_SSI)
+AC_SUBST(CONFIG_SND_IMX_TSC2102)
 AC_SUBST(CONFIG_ARCH_SA1100)
 AC_SUBST(CONFIG_L3)
 AC_SUBST(CONFIG_SND_ARMAACI)
diff -urN -X linux-2.6.18.1/Documentation/dontdiff alsa-driver-1.0.14/alsa-kernel/arm/Kconfig alsa-driver-1.0.14.mod/alsa-kernel/arm/Kconfig
--- alsa-driver-1.0.14/alsa-kernel/arm/Kconfig	2007-06-04 09:28:52.000000000 +0200
+++ alsa-driver-1.0.14.mod/alsa-kernel/arm/Kconfig	2007-08-07 11:56:04.000000000 +0200
@@ -33,4 +33,20 @@
 	  Say Y or M if you want to support any AC97 codec attached to
 	  the PXA2xx AC97 interface.
 
+config SND_IMX_SSI
+	tristate "SSI support"
+	---help---
+	  Say Y to get the SSI services API available on IMX1 platform. 
+
+config SND_IMX_TSC2102
+	tristate "IMX TSC2102 sound system"
+	depends on ARCH_IMX && IMX1_SSI
+	default N
+	select SND_PCM
+	select SND_IMX1_SSI
+	help
+	  Say Y here to include support for TSC2102 chip with audio feature.
+
+	  To compile this driver as a module, choose M here: the module
+	  will be called snd-tsc2102.
 endmenu
diff -urN -X linux-2.6.18.1/Documentation/dontdiff alsa-driver-1.0.14/alsa-kernel/arm/Makefile alsa-driver-1.0.14.mod/alsa-kernel/arm/Makefile
--- alsa-driver-1.0.14/alsa-kernel/arm/Makefile	2007-06-04 09:28:52.000000000 +0200
+++ alsa-driver-1.0.14.mod/alsa-kernel/arm/Makefile	2007-08-08 11:39:22.000000000 +0200
@@ -12,4 +12,10 @@
 snd-pxa2xx-pcm-objs		:= pxa2xx-pcm.o
 
 obj-$(CONFIG_SND_PXA2XX_AC97)	+= snd-pxa2xx-ac97.o
 snd-pxa2xx-ac97-objs		:= pxa2xx-ac97.o
+
+obj-$(CONFIG_SND_IMX_TSC2102)	+= snd-imx-tsc2102.o
+snd-imx-tsc2102-objs		:= imx-tsc2102.o
+
+obj-$(CONFIG_SND_IMX_SSI)   += snd-imx1-ssi.o
+snd-imx1-ssi-objs                := imx1-ssi.o
diff -urN -X linux-2.6.18.1/Documentation/dontdiff alsa-driver-1.0.14/arm/imx1-ssi.c alsa-driver-1.0.14.mod/arm/imx1-ssi.c
--- alsa-driver-1.0.14/arm/imx1-ssi.c	1970-01-01 01:00:00.000000000 +0100
+++ alsa-driver-1.0.14.mod/arm/imx1-ssi.c	2007-08-07 10:40:54.000000000 +0200
@@ -0,0 +1,2 @@
+#include "../alsa-kernel/arm/imx1-ssi.c"
+EXPORT_NO_SYMBOLS;
diff -urN -X linux-2.6.18.1/Documentation/dontdiff alsa-driver-1.0.14/arm/imx-tsc2102.c alsa-driver-1.0.14.mod/arm/imx-tsc2102.c
--- alsa-driver-1.0.14/arm/imx-tsc2102.c	1970-01-01 01:00:00.000000000 +0100
+++ alsa-driver-1.0.14.mod/arm/imx-tsc2102.c	2007-08-07 10:41:25.000000000 +0200
@@ -0,0 +1,2 @@
+#include "../alsa-kernel/arm/imx-tsc2102.c"
+EXPORT_NO_SYMBOLS;
--- alsa-driver-1.0.14.ref/configure.in	2007-06-04 09:28:51.000000000 +0200
+++ alsa-driver-1.0.14/configure.in	2007-09-09 22:47:35.000000000 +0200
@@ -825,6 +825,7 @@ CONFIG_L3=
 CONFIG_ARM=
 CONFIG_ARCH_SA1100=
 CONFIG_ARCH_PXA=
+CONFIG_ARCH_IMX=
 CONFIG_PPC=
 CONFIG_SPARC32=
 CONFIG_SPARC64=
@@ -918,6 +919,8 @@ int main(void) {
 	fprintf(file, "sa1100");
 #elif defined(CONFIG_ARCH_PXA)
 	fprintf(file, "pxa");
+#elif defined(CONFIG_ARCH_IMX)
+	fprintf(file, "imx");
 #elif defined(CONFIG_ARM_AMBA)
 	fprintf(file, "amba");
 #elif defined(CONFIG_PARISC)
@@ -1166,6 +1168,14 @@ EOF
     CONFIG_ARCH_SA1100=y
     test "$CONFIG_ISA" = "probe" && CONFIG_ISA=
     ;;
+  imx)
+    ARCH=arm
+    c_opts="-O2 -mapcs-32 -march=armv4 -msoft-float"
+    CONFIG_ARM=y
+    # CONFIG_L3=y
+    CONFIG_ARCH_IMX=y
+    test "$CONFIG_ISA" = "probe" && CONFIG_ISA=
+    ;;
   pxa)
     ARCH=arm
     c_opts="-O2 -mapcs-32 -march=armv4 -Wa,-mxscale -mtune=strongarm -mshort-load-bytes -msoft-float"
@@ -1334,6 +1344,7 @@ AC_SUBST(CONFIG_ALPHA)
 AC_SUBST(CONFIG_L3)
 AC_SUBST(CONFIG_ARM)
 AC_SUBST(CONFIG_ARCH_SA1100)
+AC_SUBST(CONFIG_ARCH_IMX)
 AC_SUBST(CONFIG_ARCH_PXA)
 AC_SUBST(CONFIG_PPC)
 AC_SUBST(CONFIG_SPARC32)
diff -urN -X linux-2.6.18.1/Documentation/dontdiff alsa-driver-1.0.14/toplevel.config.in alsa-driver-1.0.14.mod/toplevel.config.in
--- alsa-driver-1.0.14/toplevel.config.in	2007-06-04 09:28:55.000000000 +0200
+++ alsa-driver-1.0.14.mod/toplevel.config.in	2007-08-06 20:02:26.000000000 +0200
@@ -184,6 +184,8 @@
 CONFIG_SND_AOA_SOUNDBUS_I2S=@CONFIG_SND_AOA_SOUNDBUS_I2S@
 CONFIG_ARM=@CONFIG_ARM@
 CONFIG_SND_SA11XX_UDA1341=@CONFIG_SND_SA11XX_UDA1341@
+CONFIG_SND_IMX_SSI=@CONFIG_SND_IMX_SSI@
+CONFIG_SND_IMX_TSC2102=@CONFIG_SND_IMX_TSC2102@
 CONFIG_ARCH_SA1100=@CONFIG_ARCH_SA1100@
 CONFIG_L3=@CONFIG_L3@
 CONFIG_SND_ARMAACI=@CONFIG_SND_ARMAACI@
diff -purN alsa-driver-1.0.14.ref/include/config1.h.in alsa-driver-1.0.14/include/config1.h.in
--- alsa-driver-1.0.14.ref/include/config1.h.in	2007-06-04 09:28:55.000000000 +0200
+++ alsa-driver-1.0.14/include/config1.h.in	2007-09-09 21:37:12.000000000 +0200
@@ -188,6 +188,8 @@
 #undef CONFIG_L3_MODULE
 #undef CONFIG_SND_ARMAACI_MODULE
 #undef CONFIG_ARM_AMBA_MODULE
+#undef CONFIG_SND_IMX_SSI_MODULE
+#undef CONFIG_SND_IMX_TSC2102_MODULE
 #undef CONFIG_SND_PXA2XX_PCM_MODULE
 #undef CONFIG_SND_PXA2XX_AC97_MODULE
 #undef CONFIG_ARCH_PXA_MODULE
diff -purN alsa-driver-1.0.14.ref/scripts/ksync.conf alsa-driver-1.0.14/scripts/ksync.conf
--- alsa-driver-1.0.14.ref/scripts/ksync.conf	2007-06-04 09:28:54.000000000 +0200
+++ alsa-driver-1.0.14/scripts/ksync.conf	2007-09-09 21:01:42.000000000 +0200
@@ -49,6 +49,8 @@ comment.map.alsa-driver::= \
 	/arm/devdma.[ch]:"ARM DMA routines" \
 	/arm/aaci.[ch]:"ARM AACI PL041 driver" \
 	/arm/sa11xx-uda1341.c:"SA11xx UDA1341 driver" \
+	/arm/imx1-ssi.*:"ARM imx ssi driver" \
+	/arm/imx-tsc2102.*:"ARM imx driver" \
 	/arm/pxa2xx.*:"ARM PXA2XX driver" \
 	/arm/s3c24xx-iis.[ch]:"ARM S3C24XX IIS driver" \
 	/arm:ERROR \

