--- cygwin/gdb.mk	2006-10-25 08:40:48.859375000 +0200
+++ ../../buildroot/toolchain/gdb/gdb.mk	2006-10-20 09:15:04.000000000 +0200
@@ -192,7 +192,7 @@ $(GDB_HOST_DIR)/.configured: $(GDB_DIR)/
 
 $(GDB_HOST_DIR)/gdb/gdb: $(GDB_HOST_DIR)/.configured
 	$(MAKE) -C $(GDB_HOST_DIR)
-	strip $(GDB_HOST_DIR)/gdb/gdb
+	strip $(GDB_HOST_DIR)/gdb/gdb.exe
 
 $(TARGET_CROSS)gdb: $(GDB_HOST_DIR)/gdb/gdb
 	install -c $(GDB_HOST_DIR)/gdb/gdb $(TARGET_CROSS)gdb
