--- buildroot/package/sed/sed.mk	2006-09-05 19:48:33.015625000 +0200
+++ buildroot/package/sed/sed.mk	2006-09-16 19:50:59.203125000 +0200
@@ -66,7 +66,7 @@ use-sed-host-binary:
 	    if [ -x /bin/sed ]; then SED="/bin/sed"; fi; fi; \
 	    mkdir -p $(HOST_SED_DIR)/bin; \
 	    rm -f $(HOST_SED_DIR)/$(SED_TARGET_BINARY); \
-	    ln -s $$SED $(HOST_SED_DIR)/$(SED_TARGET_BINARY)
+	    ln -s -f $$SED $(HOST_SED_DIR)/$(SED_TARGET_BINARY)
 
 host-sed: $(HOST_SED_TARGET)
 
