diff -purN linux-2.6.12-ref/arch/arm/mach-imx/scb9328.c linux-2.6.12/arch/arm/mach-imx/scb9328.c
--- linux-2.6.12-ref/arch/arm/mach-imx/scb9328.c	2006-01-09 17:13:59.000000000 +0100
+++ linux-2.6.12/arch/arm/mach-imx/scb9328.c	2006-01-09 21:28:52.000000000 +0100
@@ -34,6 +34,8 @@
 #include <linux/usb_isp116x.h>
 #endif
 
+#include <asm/arch/imxfb.h>
+
 #define TYPE_NONE	0
 #define TYPE_EVK9328	1
 #define TYPE_EVK9328L	2
@@ -246,6 +248,37 @@ static struct platform_device serial_dev
 	},
 };
 
+static void apf9328_lcd_power(int on)
+{
+    printk("apf9328_lcd_power: %d\n", on);
+    /* use Port_D CONTRAST pin to control LCD power down mode */
+    if(on)
+        DR(3) &= 0xFFFFF7FF;    
+    else
+        DR(3) |= 0x00000800;
+}
+
+static struct imxfb_mach_info apf9328_fb_info __initdata = {
+    .pixclock   = 62500,    
+    .bpp        = 16,
+    .xres       = 240,      
+    .yres       = 320,
+
+    .hsync_len  = 9,        
+    .vsync_len  = 2,
+    .left_margin    = 9,       
+    .upper_margin   = 2,
+    .right_margin   = 27,        
+    .lower_margin   = 4,
+
+    .pcr        = PCR_TFT | PCR_COLOR | PCR_PBSIZ_8 | PCR_BPIX_16 | PCR_CLKPOL |
+                  PCR_SCLKIDLE | PCR_SCLK_SEL | PCR_PCD(2),
+
+    .lcd_power = apf9328_lcd_power,
+};
+
+
+
 static struct platform_device *devices[] __initdata = {
 	&serial_device,
 #ifdef CONFIG_DM9000
@@ -261,6 +294,15 @@ static int __init scb9328_init(void)
 /// ej	//evk9328_init_leds();
 	UFCR(IMX_UART1_BASE) = 0xa81;
 	UFCR(IMX_UART2_BASE) = 0xa81;
+ 
+	set_imx_fb_info(&apf9328_fb_info);
+
+	DR(3) |= 0x00000800; // essai Nico ...
+	DDIR(3) |= 0x7FFFFFC0;
+        GIUS(3) &= 0x8000003F;
+    	GPR(3) &= 0x8000003F;
+
+
 	return platform_add_devices(devices, ARRAY_SIZE(devices));
 }
 
